// @openrouter/sdk/esm/models/baseinputsunion.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 7237afe9aafd
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { InputAudio$inboundSchema } from "./inputaudio.js";
import { InputFile$inboundSchema } from "./inputfile.js";
import { InputImage$inboundSchema } from "./inputimage.js";
import { InputText$inboundSchema } from "./inputtext.js";
import { OutputItemImageGenerationCall$inboundSchema, } from "./outputitemimagegenerationcall.js";
import { OutputMessage$inboundSchema } from "./outputmessage.js";
import { ToolCallStatusEnum$inboundSchema, } from "./toolcallstatusenum.js";
export const BaseInputsTypeFunctionCall = {
    FunctionCall: "function_call",
};
export const BaseInputsTypeFunctionCallOutput = {
    FunctionCallOutput: "function_call_output",
};
export const BaseInputsTypeMessage2 = {
    Message: "message",
};
export const BaseInputsRoleDeveloper2 = {
    Developer: "developer",
};
export const BaseInputsRoleSystem2 = {
    System: "system",
};
export const BaseInputsRoleUser2 = {
    User: "user",
};
export const BaseInputsTypeMessage1 = {
    Message: "message",
};
export const BaseInputsRoleDeveloper1 = {
    Developer: "developer",
};
export const BaseInputsRoleAssistant = {
    Assistant: "assistant",
};
export const BaseInputsRoleSystem1 = {
    System: "system",
};
export const BaseInputsRoleUser1 = {
    User: "user",
};
export const BaseInputsPhaseFinalAnswer = {
    FinalAnswer: "final_answer",
};
export const BaseInputsPhaseCommentary = {
    Commentary: "commentary",
};
/** @internal */
export const BaseInputsTypeFunctionCall$inboundSchema = z.enum(BaseInputsTypeFunctionCall);
/** @internal */
export const BaseInputsFunctionCall$inboundSchema = z.object({
    type: BaseInputsTypeFunctionCall$inboundSchema,
    call_id: z.string(),
    name: z.string(),
    arguments: z.string(),
    id: z.string().optional(),
    status: z.nullable(ToolCallStatusEnum$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "call_id": "callId",
    });
});
export function baseInputsFunctionCallFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BaseInputsFunctionCall$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BaseInputsFunctionCall' from JSON`);
}
/** @internal */
export const BaseInputsTypeFunctionCallOutput$inboundSchema = z.enum(BaseInputsTypeFunctionCallOutput);
/** @internal */
export const BaseInputsOutput1$inboundSchema = z.union([
    InputText$inboundSchema,
    InputImage$inboundSchema.and(z.object({ type: z.literal("input_image") })),
    InputFile$inboundSchema,
]);
export function baseInputsOutput1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => BaseInputsOutput1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BaseInputsOutput1' from JSON`);
}
/** @internal */
export const BaseInputsOutput2$inboundSchema = z.union([
    z.string(),
    z.array(z.union([
        InputText$inboundSchema,
        InputImage$inboundSchema.and(z.object({ type: z.literal("input_image") })),
        InputFile$inboundSchema,
    ])),
]);
export function baseInputsOutput2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => BaseInputsOutput2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BaseInputsOutput2' from JSON`);
}
/** @internal */
export const BaseInputsFunctionCallOutput$inboundSchema = z.object({
    type: BaseInputsTypeFunctionCallOutput$inboundSchema,
    id: z.nullable(z.string()).optional(),
    call_id: z.string(),
    output: z.union([
        z.string(),
        z.array(z.union([
            InputText$inboundSchema,
            InputImage$inboundSchema.and(z.object({ type: z.literal("input_image") })),
            InputFile$inboundSchema,
        ])),
    ]),
    status: z.nullable(ToolCallStatusEnum$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "call_id": "callId",
    });
});
export function baseInputsFunctionCallOutputFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BaseInputsFunctionCallOutput$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BaseInputsFunctionCallOutput' from JSON`);
}
/** @internal */
export const BaseInputsTypeMessage2$inboundSchema = z.enum(BaseInputsTypeMessage2);
/** @internal */
export const BaseInputsRoleDeveloper2$inboundSchema = z.enum(BaseInputsRoleDeveloper2);
/** @internal */
export const BaseInputsRoleSystem2$inboundSchema = z.enum(BaseInputsRoleSystem2);
/** @internal */
export const BaseInputsRoleUser2$inboundSchema = z.enum(BaseInputsRoleUser2);
/** @internal */
export const BaseInputsRoleUnion2$inboundSchema = z.union([
    BaseInputsRoleUser2$inboundSchema,
    BaseInputsRoleSystem2$inboundSchema,
    BaseInputsRoleDeveloper2$inboundSchema,
]);
export function baseInputsRoleUnion2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => BaseInputsRoleUnion2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BaseInputsRoleUnion2' from JSON`);
}
/** @internal */
export const BaseInputsContent3$inboundSchema = z.union([
    InputText$inboundSchema,
    InputImage$inboundSchema.and(z.object({ type: z.literal("input_image") })),
    InputFile$inboundSchema,
    InputAudio$inboundSchema,
]);
export function baseInputsContent3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => BaseInputsContent3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BaseInputsContent3' from JSON`);
}
/** @internal */
export const BaseInputsMessage2$inboundSchema = z.object({
    id: z.string(),
    type: BaseInputsTypeMessage2$inboundSchema.optional(),
    role: z.union([
        BaseInputsRoleUser2$inboundSchema,
        BaseInputsRoleSystem2$inboundSchema,
        BaseInputsRoleDeveloper2$inboundSchema,
    ]),
    content: z.array(z.union([
        InputText$inboundSchema,
        InputImage$inboundSchema.and(z.object({ type: z.literal("input_image") })),
        InputFile$inboundSchema,
        InputAudio$inboundSchema,
    ])),
});
export function baseInputsMessage2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => BaseInputsMessage2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BaseInputsMessage2' from JSON`);
}
/** @internal */
export const BaseInputsTypeMessage1$inboundSchema = z.enum(BaseInputsTypeMessage1);
/** @internal */
export const BaseInputsRoleDeveloper1$inboundSchema = z.enum(BaseInputsRoleDeveloper1);
/** @internal */
export const BaseInputsRoleAssistant$inboundSchema = z.enum(BaseInputsRoleAssistant);
/** @internal */
export const BaseInputsRoleSystem1$inboundSchema = z.enum(BaseInputsRoleSystem1);
/** @internal */
export const BaseInputsRoleUser1$inboundSchema = z.enum(BaseInputsRoleUser1);
/** @internal */
export const BaseInputsRoleUnion1$inboundSchema = z.union([
    BaseInputsRoleUser1$inboundSchema,
    BaseInputsRoleSystem1$inboundSchema,
    BaseInputsRoleAssistant$inboundSchema,
    BaseInputsRoleDeveloper1$inboundSchema,
]);
export function baseInputsRoleUnion1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => BaseInputsRoleUnion1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BaseInputsRoleUnion1' from JSON`);
}
/** @internal */
export const BaseInputsContent1$inboundSchema = z.union([
    InputText$inboundSchema,
    InputImage$inboundSchema.and(z.object({ type: z.literal("input_image") })),
    InputFile$inboundSchema,
    InputAudio$inboundSchema,
]);
export function baseInputsContent1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => BaseInputsContent1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BaseInputsContent1' from JSON`);
}
/** @internal */
export const BaseInputsContent2$inboundSchema = z.union([
    z.array(z.union([
        InputText$inboundSchema,
        InputImage$inboundSchema.and(z.object({ type: z.literal("input_image") })),
        InputFile$inboundSchema,
        InputAudio$inboundSchema,
    ])),
    z.string(),
]);
export function baseInputsContent2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => BaseInputsContent2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BaseInputsContent2' from JSON`);
}
/** @internal */
export const BaseInputsPhaseFinalAnswer$inboundSchema = z.enum(BaseInputsPhaseFinalAnswer);
/** @internal */
export const BaseInputsPhaseCommentary$inboundSchema = z.enum(BaseInputsPhaseCommentary);
/** @internal */
export const BaseInputsPhaseUnion$inboundSchema = z.union([
    BaseInputsPhaseCommentary$inboundSchema,
    BaseInputsPhaseFinalAnswer$inboundSchema,
    z.any(),
]);
export function baseInputsPhaseUnionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BaseInputsPhaseUnion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BaseInputsPhaseUnion' from JSON`);
}
/** @internal */
export const BaseInputsMessage1$inboundSchema = z.object({
    type: BaseInputsTypeMessage1$inboundSchema.optional(),
    role: z.union([
        BaseInputsRoleUser1$inboundSchema,
        BaseInputsRoleSystem1$inboundSchema,
        BaseInputsRoleAssistant$inboundSchema,
        BaseInputsRoleDeveloper1$inboundSchema,
    ]),
    content: z.union([
        z.array(z.union([
            InputText$inboundSchema,
            InputImage$inboundSchema.and(z.object({ type: z.literal("input_image") })),
            InputFile$inboundSchema,
            InputAudio$inboundSchema,
        ])),
        z.string(),
    ]),
    phase: z.nullable(z.union([
        BaseInputsPhaseCommentary$inboundSchema,
        BaseInputsPhaseFinalAnswer$inboundSchema,
        z.any(),
    ])).optional(),
});
export function baseInputsMessage1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => BaseInputsMessage1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BaseInputsMessage1' from JSON`);
}
/** @internal */
export const BaseInputsUnion1$inboundSchema = z.union([
    z.lazy(() => BaseInputsFunctionCall$inboundSchema),
    OutputMessage$inboundSchema,
    z.lazy(() => BaseInputsMessage2$inboundSchema),
    z.lazy(() => BaseInputsFunctionCallOutput$inboundSchema),
    OutputItemImageGenerationCall$inboundSchema,
    z.lazy(() => BaseInputsMessage1$inboundSchema),
]);
export function baseInputsUnion1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => BaseInputsUnion1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BaseInputsUnion1' from JSON`);
}
/** @internal */
export const BaseInputsUnion$inboundSchema = z.union([
    z.string(),
    z.array(z.union([
        z.lazy(() => BaseInputsFunctionCall$inboundSchema),
        OutputMessage$inboundSchema,
        z.lazy(() => BaseInputsMessage2$inboundSchema),
        z.lazy(() => BaseInputsFunctionCallOutput$inboundSchema),
        OutputItemImageGenerationCall$inboundSchema,
        z.lazy(() => BaseInputsMessage1$inboundSchema),
    ])),
    z.any(),
]);
export function baseInputsUnionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BaseInputsUnion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BaseInputsUnion' from JSON`);
}
