// @openrouter/sdk/esm/models/chatcontentcachecontrol.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 2826e2627ca8
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const ChatContentCacheControlType = {
    Ephemeral: "ephemeral",
};
export const ChatContentCacheControlTtl = {
    Fivem: "5m",
    Oneh: "1h",
};
/** @internal */
export const ChatContentCacheControlType$inboundSchema = z.enum(ChatContentCacheControlType);
/** @internal */
export const ChatContentCacheControlType$outboundSchema = ChatContentCacheControlType$inboundSchema;
/** @internal */
export const ChatContentCacheControlTtl$inboundSchema = openEnums.inboundSchema(ChatContentCacheControlTtl);
/** @internal */
export const ChatContentCacheControlTtl$outboundSchema = openEnums.outboundSchema(ChatContentCacheControlTtl);
/** @internal */
export const ChatContentCacheControl$inboundSchema = z.object({
    type: ChatContentCacheControlType$inboundSchema,
    ttl: ChatContentCacheControlTtl$inboundSchema.optional(),
});
/** @internal */
export const ChatContentCacheControl$outboundSchema = z.object({
    type: ChatContentCacheControlType$outboundSchema,
    ttl: ChatContentCacheControlTtl$outboundSchema.optional(),
});
export function chatContentCacheControlToJSON(chatContentCacheControl) {
    return JSON.stringify(ChatContentCacheControl$outboundSchema.parse(chatContentCacheControl));
}
export function chatContentCacheControlFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatContentCacheControl$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatContentCacheControl' from JSON`);
}
