// @openrouter/sdk/esm/models/chatcontentaudio.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 0f458a760f42
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const ChatContentAudioType = {
    InputAudio: "input_audio",
};
/** @internal */
export const ChatContentAudioType$inboundSchema = z.enum(ChatContentAudioType);
/** @internal */
export const ChatContentAudioType$outboundSchema = ChatContentAudioType$inboundSchema;
/** @internal */
export const ChatContentAudioInputAudio$inboundSchema = z.object({
    data: z.string(),
    format: z.string(),
});
/** @internal */
export const ChatContentAudioInputAudio$outboundSchema = z.object({
    data: z.string(),
    format: z.string(),
});
export function chatContentAudioInputAudioToJSON(chatContentAudioInputAudio) {
    return JSON.stringify(ChatContentAudioInputAudio$outboundSchema.parse(chatContentAudioInputAudio));
}
export function chatContentAudioInputAudioFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatContentAudioInputAudio$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatContentAudioInputAudio' from JSON`);
}
/** @internal */
export const ChatContentAudio$inboundSchema = z.object({
    type: ChatContentAudioType$inboundSchema,
    input_audio: z.lazy(() => ChatContentAudioInputAudio$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "input_audio": "inputAudio",
    });
});
/** @internal */
export const ChatContentAudio$outboundSchema = z.object({
    type: ChatContentAudioType$outboundSchema,
    inputAudio: z.lazy(() => ChatContentAudioInputAudio$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        inputAudio: "input_audio",
    });
});
export function chatContentAudioToJSON(chatContentAudio) {
    return JSON.stringify(ChatContentAudio$outboundSchema.parse(chatContentAudio));
}
export function chatContentAudioFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatContentAudio$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatContentAudio' from JSON`);
}
