// @openrouter/sdk/esm/models/chatwebsearchservertool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: ae822f875178
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
export const ChatWebSearchServerToolTypeOpenrouterWebSearch = {
    OpenrouterWebSearch: "openrouter:web_search",
};
/**
 * Which search engine to use. "auto" (default) uses native if the provider supports it, otherwise Exa. "native" forces the provider's built-in search. "exa" forces the Exa search API. "firecrawl" uses Firecrawl (requires BYOK). "parallel" uses the Parallel search API.
 */
export const ChatWebSearchServerToolEngine = {
    Auto: "auto",
    Native: "native",
    Exa: "exa",
    Firecrawl: "firecrawl",
    Parallel: "parallel",
};
/**
 * How much context to retrieve per result. Defaults to medium (15000 chars). Only applies when using the Exa engine; ignored with native provider search.
 */
export const ChatWebSearchServerToolSearchContextSize = {
    Low: "low",
    Medium: "medium",
    High: "high",
};
export const ChatWebSearchServerToolParametersType = {
    Approximate: "approximate",
};
/** @internal */
export const ChatWebSearchServerToolTypeOpenrouterWebSearch$outboundSchema = z.enum(ChatWebSearchServerToolTypeOpenrouterWebSearch);
/** @internal */
export const ChatWebSearchServerToolEngine$outboundSchema = openEnums.outboundSchema(ChatWebSearchServerToolEngine);
/** @internal */
export const ChatWebSearchServerToolSearchContextSize$outboundSchema = openEnums.outboundSchema(ChatWebSearchServerToolSearchContextSize);
/** @internal */
export const ChatWebSearchServerToolParametersType$outboundSchema = z.enum(ChatWebSearchServerToolParametersType);
/** @internal */
export const ChatWebSearchServerToolUserLocation$outboundSchema = z.object({
    type: ChatWebSearchServerToolParametersType$outboundSchema.optional(),
    city: z.string().optional(),
    region: z.string().optional(),
    country: z.string().optional(),
    timezone: z.string().optional(),
});
export function chatWebSearchServerToolUserLocationToJSON(chatWebSearchServerToolUserLocation) {
    return JSON.stringify(ChatWebSearchServerToolUserLocation$outboundSchema.parse(chatWebSearchServerToolUserLocation));
}
/** @internal */
export const ChatWebSearchServerToolParameters$outboundSchema = z.object({
    engine: ChatWebSearchServerToolEngine$outboundSchema.optional(),
    maxResults: z.number().optional(),
    maxTotalResults: z.number().optional(),
    searchContextSize: ChatWebSearchServerToolSearchContextSize$outboundSchema
        .optional(),
    userLocation: z.lazy(() => ChatWebSearchServerToolUserLocation$outboundSchema)
        .optional(),
    allowedDomains: z.array(z.string()).optional(),
    excludedDomains: z.array(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        maxResults: "max_results",
        maxTotalResults: "max_total_results",
        searchContextSize: "search_context_size",
        userLocation: "user_location",
        allowedDomains: "allowed_domains",
        excludedDomains: "excluded_domains",
    });
});
export function chatWebSearchServerToolParametersToJSON(chatWebSearchServerToolParameters) {
    return JSON.stringify(ChatWebSearchServerToolParameters$outboundSchema.parse(chatWebSearchServerToolParameters));
}
/** @internal */
export const ChatWebSearchServerTool$outboundSchema = z.object({
    type: ChatWebSearchServerToolTypeOpenrouterWebSearch$outboundSchema,
    parameters: z.lazy(() => ChatWebSearchServerToolParameters$outboundSchema)
        .optional(),
});
export function chatWebSearchServerToolToJSON(chatWebSearchServerTool) {
    return JSON.stringify(ChatWebSearchServerTool$outboundSchema.parse(chatWebSearchServerTool));
}
