// @openrouter/sdk/esm/models/chatcontentimage.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 0903ebf8f7a0
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const ChatContentImageType = {
    ImageUrl: "image_url",
};
/**
 * Image detail level for vision models
 */
export const ChatContentImageDetail = {
    Auto: "auto",
    Low: "low",
    High: "high",
};
/** @internal */
export const ChatContentImageType$inboundSchema = z.enum(ChatContentImageType);
/** @internal */
export const ChatContentImageType$outboundSchema = ChatContentImageType$inboundSchema;
/** @internal */
export const ChatContentImageDetail$inboundSchema = openEnums.inboundSchema(ChatContentImageDetail);
/** @internal */
export const ChatContentImageDetail$outboundSchema = openEnums.outboundSchema(ChatContentImageDetail);
/** @internal */
export const ChatContentImageImageUrl$inboundSchema = z.object({
    url: z.string(),
    detail: ChatContentImageDetail$inboundSchema.optional(),
});
/** @internal */
export const ChatContentImageImageUrl$outboundSchema = z.object({
    url: z.string(),
    detail: ChatContentImageDetail$outboundSchema.optional(),
});
export function chatContentImageImageUrlToJSON(chatContentImageImageUrl) {
    return JSON.stringify(ChatContentImageImageUrl$outboundSchema.parse(chatContentImageImageUrl));
}
export function chatContentImageImageUrlFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatContentImageImageUrl$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatContentImageImageUrl' from JSON`);
}
/** @internal */
export const ChatContentImage$inboundSchema = z.object({
    type: ChatContentImageType$inboundSchema,
    image_url: z.lazy(() => ChatContentImageImageUrl$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "image_url": "imageUrl",
    });
});
/** @internal */
export const ChatContentImage$outboundSchema = z.object({
    type: ChatContentImageType$outboundSchema,
    imageUrl: z.lazy(() => ChatContentImageImageUrl$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        imageUrl: "image_url",
    });
});
export function chatContentImageToJSON(chatContentImage) {
    return JSON.stringify(ChatContentImage$outboundSchema.parse(chatContentImage));
}
export function chatContentImageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatContentImage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatContentImage' from JSON`);
}
