// @openrouter/sdk/esm/models/chatcontentitems.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 3d8c886d8084
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { ChatContentAudio$inboundSchema, ChatContentAudio$outboundSchema, } from "./chatcontentaudio.js";
import { ChatContentFile$inboundSchema, ChatContentFile$outboundSchema, } from "./chatcontentfile.js";
import { ChatContentImage$inboundSchema, ChatContentImage$outboundSchema, } from "./chatcontentimage.js";
import { ChatContentText$inboundSchema, ChatContentText$outboundSchema, } from "./chatcontenttext.js";
import { ChatContentVideo$inboundSchema, ChatContentVideo$outboundSchema, } from "./chatcontentvideo.js";
import { LegacyChatContentVideo$inboundSchema, LegacyChatContentVideo$outboundSchema, } from "./legacychatcontentvideo.js";
/** @internal */
export const ChatContentItems1$inboundSchema = z.union([
    LegacyChatContentVideo$inboundSchema,
    ChatContentVideo$inboundSchema,
]);
/** @internal */
export const ChatContentItems1$outboundSchema = z.union([
    LegacyChatContentVideo$outboundSchema,
    ChatContentVideo$outboundSchema,
]);
export function chatContentItems1ToJSON(chatContentItems1) {
    return JSON.stringify(ChatContentItems1$outboundSchema.parse(chatContentItems1));
}
export function chatContentItems1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatContentItems1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatContentItems1' from JSON`);
}
/** @internal */
export const ChatContentItems$inboundSchema = z.union([
    ChatContentText$inboundSchema,
    ChatContentImage$inboundSchema,
    ChatContentAudio$inboundSchema,
    ChatContentFile$inboundSchema,
    z.union([
        LegacyChatContentVideo$inboundSchema,
        ChatContentVideo$inboundSchema,
    ]),
]);
/** @internal */
export const ChatContentItems$outboundSchema = z.union([
    ChatContentText$outboundSchema,
    ChatContentImage$outboundSchema,
    ChatContentAudio$outboundSchema,
    ChatContentFile$outboundSchema,
    z.union([
        LegacyChatContentVideo$outboundSchema,
        ChatContentVideo$outboundSchema,
    ]),
]);
export function chatContentItemsToJSON(chatContentItems) {
    return JSON.stringify(ChatContentItems$outboundSchema.parse(chatContentItems));
}
export function chatContentItemsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatContentItems$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatContentItems' from JSON`);
}
