// @openrouter/sdk/esm/models/chatwebsearchshorthand.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: a67d6bbce478
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
export const ChatWebSearchShorthandType = {
    WebSearch: "web_search",
    WebSearchPreview: "web_search_preview",
    WebSearchPreview20250311: "web_search_preview_2025_03_11",
    WebSearch20250826: "web_search_2025_08_26",
};
/**
 * Which search engine to use. "auto" (default) uses native if the provider supports it, otherwise Exa. "native" forces the provider's built-in search. "exa" forces the Exa search API. "firecrawl" uses Firecrawl (requires BYOK). "parallel" uses the Parallel search API.
 */
export const ChatWebSearchShorthandEngine = {
    Auto: "auto",
    Native: "native",
    Exa: "exa",
    Firecrawl: "firecrawl",
    Parallel: "parallel",
};
/**
 * How much context to retrieve per result. Defaults to medium (15000 chars). Only applies when using the Exa engine; ignored with native provider search.
 */
export const ChatWebSearchShorthandSearchContextSize = {
    Low: "low",
    Medium: "medium",
    High: "high",
};
export const ChatWebSearchShorthandTypeApproximate = {
    Approximate: "approximate",
};
/**
 * Which search engine to use. "auto" (default) uses native if the provider supports it, otherwise Exa. "native" forces the provider's built-in search. "exa" forces the Exa search API. "firecrawl" uses Firecrawl (requires BYOK). "parallel" uses the Parallel search API.
 */
export const ChatWebSearchShorthandParametersEngine = {
    Auto: "auto",
    Native: "native",
    Exa: "exa",
    Firecrawl: "firecrawl",
    Parallel: "parallel",
};
/**
 * How much context to retrieve per result. Defaults to medium (15000 chars). Only applies when using the Exa engine; ignored with native provider search.
 */
export const ChatWebSearchShorthandParametersSearchContextSize = {
    Low: "low",
    Medium: "medium",
    High: "high",
};
export const ChatWebSearchShorthandParametersType = {
    Approximate: "approximate",
};
/** @internal */
export const ChatWebSearchShorthandType$outboundSchema = openEnums.outboundSchema(ChatWebSearchShorthandType);
/** @internal */
export const ChatWebSearchShorthandEngine$outboundSchema = openEnums.outboundSchema(ChatWebSearchShorthandEngine);
/** @internal */
export const ChatWebSearchShorthandSearchContextSize$outboundSchema = openEnums.outboundSchema(ChatWebSearchShorthandSearchContextSize);
/** @internal */
export const ChatWebSearchShorthandTypeApproximate$outboundSchema = z.enum(ChatWebSearchShorthandTypeApproximate);
/** @internal */
export const ChatWebSearchShorthandUserLocation$outboundSchema = z.object({
    type: ChatWebSearchShorthandTypeApproximate$outboundSchema.optional(),
    city: z.string().optional(),
    region: z.string().optional(),
    country: z.string().optional(),
    timezone: z.string().optional(),
});
export function chatWebSearchShorthandUserLocationToJSON(chatWebSearchShorthandUserLocation) {
    return JSON.stringify(ChatWebSearchShorthandUserLocation$outboundSchema.parse(chatWebSearchShorthandUserLocation));
}
/** @internal */
export const ChatWebSearchShorthandParametersEngine$outboundSchema = openEnums.outboundSchema(ChatWebSearchShorthandParametersEngine);
/** @internal */
export const ChatWebSearchShorthandParametersSearchContextSize$outboundSchema = openEnums.outboundSchema(ChatWebSearchShorthandParametersSearchContextSize);
/** @internal */
export const ChatWebSearchShorthandParametersType$outboundSchema = z.enum(ChatWebSearchShorthandParametersType);
/** @internal */
export const ChatWebSearchShorthandParametersUserLocation$outboundSchema = z.object({
    type: ChatWebSearchShorthandParametersType$outboundSchema.optional(),
    city: z.string().optional(),
    region: z.string().optional(),
    country: z.string().optional(),
    timezone: z.string().optional(),
});
export function chatWebSearchShorthandParametersUserLocationToJSON(chatWebSearchShorthandParametersUserLocation) {
    return JSON.stringify(ChatWebSearchShorthandParametersUserLocation$outboundSchema.parse(chatWebSearchShorthandParametersUserLocation));
}
/** @internal */
export const ChatWebSearchShorthandParameters$outboundSchema = z.object({
    engine: ChatWebSearchShorthandParametersEngine$outboundSchema.optional(),
    maxResults: z.number().optional(),
    maxTotalResults: z.number().optional(),
    searchContextSize: ChatWebSearchShorthandParametersSearchContextSize$outboundSchema.optional(),
    userLocation: z.lazy(() => ChatWebSearchShorthandParametersUserLocation$outboundSchema).optional(),
    allowedDomains: z.array(z.string()).optional(),
    excludedDomains: z.array(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        maxResults: "max_results",
        maxTotalResults: "max_total_results",
        searchContextSize: "search_context_size",
        userLocation: "user_location",
        allowedDomains: "allowed_domains",
        excludedDomains: "excluded_domains",
    });
});
export function chatWebSearchShorthandParametersToJSON(chatWebSearchShorthandParameters) {
    return JSON.stringify(ChatWebSearchShorthandParameters$outboundSchema.parse(chatWebSearchShorthandParameters));
}
/** @internal */
export const ChatWebSearchShorthand$outboundSchema = z.object({
    type: ChatWebSearchShorthandType$outboundSchema,
    engine: ChatWebSearchShorthandEngine$outboundSchema.optional(),
    maxResults: z.number().optional(),
    maxTotalResults: z.number().optional(),
    searchContextSize: ChatWebSearchShorthandSearchContextSize$outboundSchema
        .optional(),
    userLocation: z.lazy(() => ChatWebSearchShorthandUserLocation$outboundSchema)
        .optional(),
    allowedDomains: z.array(z.string()).optional(),
    excludedDomains: z.array(z.string()).optional(),
    parameters: z.lazy(() => ChatWebSearchShorthandParameters$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        maxResults: "max_results",
        maxTotalResults: "max_total_results",
        searchContextSize: "search_context_size",
        userLocation: "user_location",
        allowedDomains: "allowed_domains",
        excludedDomains: "excluded_domains",
    });
});
export function chatWebSearchShorthandToJSON(chatWebSearchShorthand) {
    return JSON.stringify(ChatWebSearchShorthand$outboundSchema.parse(chatWebSearchShorthand));
}
