// @openrouter/sdk/esm/models/chatassistantmessage.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: d8ae826b4712
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ChatAssistantImages$inboundSchema, ChatAssistantImages$outboundSchema, } from "./chatassistantimages.js";
import { ChatAudioOutput$inboundSchema, ChatAudioOutput$outboundSchema, } from "./chataudiooutput.js";
import { ChatContentItems$inboundSchema, ChatContentItems$outboundSchema, } from "./chatcontentitems.js";
import { ChatToolCall$inboundSchema, ChatToolCall$outboundSchema, } from "./chattoolcall.js";
import { ReasoningDetailUnion$inboundSchema, ReasoningDetailUnion$outboundSchema, } from "./reasoningdetailunion.js";
export const ChatAssistantMessageRole = {
    Assistant: "assistant",
};
/** @internal */
export const ChatAssistantMessageRole$inboundSchema = z.enum(ChatAssistantMessageRole);
/** @internal */
export const ChatAssistantMessageRole$outboundSchema = ChatAssistantMessageRole$inboundSchema;
/** @internal */
export const ChatAssistantMessageContent$inboundSchema = z.union([z.string(), z.array(ChatContentItems$inboundSchema), z.any()]);
/** @internal */
export const ChatAssistantMessageContent$outboundSchema = z.union([z.string(), z.array(ChatContentItems$outboundSchema), z.any()]);
export function chatAssistantMessageContentToJSON(chatAssistantMessageContent) {
    return JSON.stringify(ChatAssistantMessageContent$outboundSchema.parse(chatAssistantMessageContent));
}
export function chatAssistantMessageContentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatAssistantMessageContent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatAssistantMessageContent' from JSON`);
}
/** @internal */
export const ChatAssistantMessage$inboundSchema = z.object({
    role: ChatAssistantMessageRole$inboundSchema,
    content: z.nullable(z.union([z.string(), z.array(ChatContentItems$inboundSchema), z.any()])).optional(),
    name: z.string().optional(),
    tool_calls: z.array(ChatToolCall$inboundSchema).optional(),
    refusal: z.nullable(z.string()).optional(),
    reasoning: z.nullable(z.string()).optional(),
    reasoning_details: z.array(ReasoningDetailUnion$inboundSchema).optional(),
    images: z.array(ChatAssistantImages$inboundSchema).optional(),
    audio: ChatAudioOutput$inboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        "tool_calls": "toolCalls",
        "reasoning_details": "reasoningDetails",
    });
});
/** @internal */
export const ChatAssistantMessage$outboundSchema = z.object({
    role: ChatAssistantMessageRole$outboundSchema,
    content: z.nullable(z.union([z.string(), z.array(ChatContentItems$outboundSchema), z.any()])).optional(),
    name: z.string().optional(),
    toolCalls: z.array(ChatToolCall$outboundSchema).optional(),
    refusal: z.nullable(z.string()).optional(),
    reasoning: z.nullable(z.string()).optional(),
    reasoningDetails: z.array(ReasoningDetailUnion$outboundSchema).optional(),
    images: z.array(ChatAssistantImages$outboundSchema).optional(),
    audio: ChatAudioOutput$outboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        toolCalls: "tool_calls",
        reasoningDetails: "reasoning_details",
    });
});
export function chatAssistantMessageToJSON(chatAssistantMessage) {
    return JSON.stringify(ChatAssistantMessage$outboundSchema.parse(chatAssistantMessage));
}
export function chatAssistantMessageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatAssistantMessage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatAssistantMessage' from JSON`);
}
