// @openrouter/sdk/esm/models/chattoolcall.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: f15fee305732
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
export const ChatToolCallType = {
    Function: "function",
};
/** @internal */
export const ChatToolCallType$inboundSchema = z.enum(ChatToolCallType);
/** @internal */
export const ChatToolCallType$outboundSchema = ChatToolCallType$inboundSchema;
/** @internal */
export const ChatToolCallFunction$inboundSchema = z.object({
    name: z.string(),
    arguments: z.string(),
});
/** @internal */
export const ChatToolCallFunction$outboundSchema = z.object({
    name: z.string(),
    arguments: z.string(),
});
export function chatToolCallFunctionToJSON(chatToolCallFunction) {
    return JSON.stringify(ChatToolCallFunction$outboundSchema.parse(chatToolCallFunction));
}
export function chatToolCallFunctionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatToolCallFunction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatToolCallFunction' from JSON`);
}
/** @internal */
export const ChatToolCall$inboundSchema = z
    .object({
    id: z.string(),
    type: ChatToolCallType$inboundSchema,
    function: z.lazy(() => ChatToolCallFunction$inboundSchema),
});
/** @internal */
export const ChatToolCall$outboundSchema = z.object({
    id: z.string(),
    type: ChatToolCallType$outboundSchema,
    function: z.lazy(() => ChatToolCallFunction$outboundSchema),
});
export function chatToolCallToJSON(chatToolCall) {
    return JSON.stringify(ChatToolCall$outboundSchema.parse(chatToolCall));
}
export function chatToolCallFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ChatToolCall$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ChatToolCall' from JSON`);
}
