// @openrouter/sdk/esm/models/computeruseservertool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: e6059da59a69
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const Environment = {
    Windows: "windows",
    Mac: "mac",
    Linux: "linux",
    Ubuntu: "ubuntu",
    Browser: "browser",
};
/** @internal */
export const Environment$inboundSchema = openEnums.inboundSchema(Environment);
/** @internal */
export const Environment$outboundSchema = openEnums.outboundSchema(Environment);
/** @internal */
export const ComputerUseServerTool$inboundSchema = z.object({
    type: z.literal("computer_use_preview"),
    display_height: z.number(),
    display_width: z.number(),
    environment: Environment$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "display_height": "displayHeight",
        "display_width": "displayWidth",
    });
});
/** @internal */
export const ComputerUseServerTool$outboundSchema = z.object({
    type: z.literal("computer_use_preview"),
    displayHeight: z.number(),
    displayWidth: z.number(),
    environment: Environment$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        displayHeight: "display_height",
        displayWidth: "display_width",
    });
});
export function computerUseServerToolToJSON(computerUseServerTool) {
    return JSON.stringify(ComputerUseServerTool$outboundSchema.parse(computerUseServerTool));
}
export function computerUseServerToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ComputerUseServerTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ComputerUseServerTool' from JSON`);
}
