// @openrouter/sdk/esm/models/contentpartdoneevent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 943cfab7678a
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { OpenAIResponsesRefusalContent$inboundSchema, } from "./openairesponsesrefusalcontent.js";
import { ReasoningTextContent$inboundSchema, } from "./reasoningtextcontent.js";
import { ResponseOutputText$inboundSchema, } from "./responseoutputtext.js";
/** @internal */
export const ContentPartDoneEventPart$inboundSchema = z.union([
    ResponseOutputText$inboundSchema,
    ReasoningTextContent$inboundSchema.and(z.object({ type: z.literal("reasoning_text") })),
    OpenAIResponsesRefusalContent$inboundSchema,
]);
export function contentPartDoneEventPartFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ContentPartDoneEventPart$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ContentPartDoneEventPart' from JSON`);
}
/** @internal */
export const ContentPartDoneEvent$inboundSchema = z.object({
    type: z.literal("response.content_part.done"),
    output_index: z.number(),
    item_id: z.string(),
    content_index: z.number(),
    part: z.union([
        ResponseOutputText$inboundSchema,
        ReasoningTextContent$inboundSchema.and(z.object({ type: z.literal("reasoning_text") })),
        OpenAIResponsesRefusalContent$inboundSchema,
    ]),
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "output_index": "outputIndex",
        "item_id": "itemId",
        "content_index": "contentIndex",
        "sequence_number": "sequenceNumber",
    });
});
export function contentPartDoneEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ContentPartDoneEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ContentPartDoneEvent' from JSON`);
}
