// @openrouter/sdk/esm/models/formatjsonschemaconfig.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: dbec4c78c132
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const FormatJsonSchemaConfig$inboundSchema = z.object({
    type: z.literal("json_schema"),
    name: z.string(),
    description: z.string().optional(),
    strict: z.nullable(z.boolean()).optional(),
    schema: z.record(z.string(), z.nullable(z.any())),
});
/** @internal */
export const FormatJsonSchemaConfig$outboundSchema = z.object({
    type: z.literal("json_schema"),
    name: z.string(),
    description: z.string().optional(),
    strict: z.nullable(z.boolean()).optional(),
    schema: z.record(z.string(), z.nullable(z.any())),
});
export function formatJsonSchemaConfigToJSON(formatJsonSchemaConfig) {
    return JSON.stringify(FormatJsonSchemaConfig$outboundSchema.parse(formatJsonSchemaConfig));
}
export function formatJsonSchemaConfigFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FormatJsonSchemaConfig$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FormatJsonSchemaConfig' from JSON`);
}
