// @openrouter/sdk/esm/models/imagegencallpartialimageevent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: db1f02788378
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const ImageGenCallPartialImageEvent$inboundSchema = z.object({
    type: z.literal("response.image_generation_call.partial_image"),
    item_id: z.string(),
    output_index: z.number(),
    sequence_number: z.number(),
    partial_image_b64: z.string(),
    partial_image_index: z.number(),
}).transform((v) => {
    return remap$(v, {
        "item_id": "itemId",
        "output_index": "outputIndex",
        "sequence_number": "sequenceNumber",
        "partial_image_b64": "partialImageB64",
        "partial_image_index": "partialImageIndex",
    });
});
export function imageGenCallPartialImageEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImageGenCallPartialImageEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImageGenCallPartialImageEvent' from JSON`);
}
