// @openrouter/sdk/esm/models/inputsunion.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 3aff2165a17d
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { EasyInputMessage$outboundSchema, } from "./easyinputmessage.js";
import { FunctionCallItem$outboundSchema, } from "./functioncallitem.js";
import { FunctionCallOutputItem$outboundSchema, } from "./functioncalloutputitem.js";
import { InputMessageItem$outboundSchema, } from "./inputmessageitem.js";
import { OpenAIResponsesRefusalContent$outboundSchema, } from "./openairesponsesrefusalcontent.js";
import { OutputDatetimeItem$outboundSchema, } from "./outputdatetimeitem.js";
import { OutputFileSearchCallItem$outboundSchema, } from "./outputfilesearchcallitem.js";
import { OutputFunctionCallItem$outboundSchema, } from "./outputfunctioncallitem.js";
import { OutputImageGenerationCallItem$outboundSchema, } from "./outputimagegenerationcallitem.js";
import { OutputServerToolItem$outboundSchema, } from "./outputservertoolitem.js";
import { OutputWebSearchCallItem$outboundSchema, } from "./outputwebsearchcallitem.js";
import { ReasoningItem$outboundSchema, } from "./reasoningitem.js";
import { ReasoningSummaryText$outboundSchema, } from "./reasoningsummarytext.js";
import { ReasoningTextContent$outboundSchema, } from "./reasoningtextcontent.js";
import { ResponseOutputText$outboundSchema, } from "./responseoutputtext.js";
export const InputsTypeReasoning = {
    Reasoning: "reasoning",
};
export const InputsStatusInProgress2 = {
    InProgress: "in_progress",
};
export const InputsStatusIncomplete2 = {
    Incomplete: "incomplete",
};
export const InputsStatusCompleted2 = {
    Completed: "completed",
};
/**
 * The format of the reasoning content
 */
export const InputsFormat = {
    Unknown: "unknown",
    OpenaiResponsesV1: "openai-responses-v1",
    AzureOpenaiResponsesV1: "azure-openai-responses-v1",
    XaiResponsesV1: "xai-responses-v1",
    AnthropicClaudeV1: "anthropic-claude-v1",
    GoogleGeminiV1: "google-gemini-v1",
};
export const InputsRole = {
    Assistant: "assistant",
};
export const InputsTypeMessage = {
    Message: "message",
};
export const InputsStatusInProgress1 = {
    InProgress: "in_progress",
};
export const InputsStatusIncomplete1 = {
    Incomplete: "incomplete",
};
export const InputsStatusCompleted1 = {
    Completed: "completed",
};
export const InputsPhaseFinalAnswer = {
    FinalAnswer: "final_answer",
};
export const InputsPhaseCommentary = {
    Commentary: "commentary",
};
/** @internal */
export const InputsTypeReasoning$outboundSchema = z.enum(InputsTypeReasoning);
/** @internal */
export const InputsStatusInProgress2$outboundSchema = z.enum(InputsStatusInProgress2);
/** @internal */
export const InputsStatusIncomplete2$outboundSchema = z.enum(InputsStatusIncomplete2);
/** @internal */
export const InputsStatusCompleted2$outboundSchema = z.enum(InputsStatusCompleted2);
/** @internal */
export const InputsStatusUnion2$outboundSchema = z.union([
    InputsStatusCompleted2$outboundSchema,
    InputsStatusIncomplete2$outboundSchema,
    InputsStatusInProgress2$outboundSchema,
]);
export function inputsStatusUnion2ToJSON(inputsStatusUnion2) {
    return JSON.stringify(InputsStatusUnion2$outboundSchema.parse(inputsStatusUnion2));
}
/** @internal */
export const InputsFormat$outboundSchema = openEnums.outboundSchema(InputsFormat);
/** @internal */
export const InputsReasoning$outboundSchema = z.object({
    type: InputsTypeReasoning$outboundSchema,
    id: z.string(),
    content: z.nullable(z.array(ReasoningTextContent$outboundSchema)).optional(),
    summary: z.nullable(z.array(ReasoningSummaryText$outboundSchema)),
    encryptedContent: z.nullable(z.string()).optional(),
    status: z.union([
        InputsStatusCompleted2$outboundSchema,
        InputsStatusIncomplete2$outboundSchema,
        InputsStatusInProgress2$outboundSchema,
    ]).optional(),
    signature: z.nullable(z.string()).optional(),
    format: z.nullable(InputsFormat$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        encryptedContent: "encrypted_content",
    });
});
export function inputsReasoningToJSON(inputsReasoning) {
    return JSON.stringify(InputsReasoning$outboundSchema.parse(inputsReasoning));
}
/** @internal */
export const InputsRole$outboundSchema = z.enum(InputsRole);
/** @internal */
export const InputsTypeMessage$outboundSchema = z.enum(InputsTypeMessage);
/** @internal */
export const InputsStatusInProgress1$outboundSchema = z.enum(InputsStatusInProgress1);
/** @internal */
export const InputsStatusIncomplete1$outboundSchema = z.enum(InputsStatusIncomplete1);
/** @internal */
export const InputsStatusCompleted1$outboundSchema = z.enum(InputsStatusCompleted1);
/** @internal */
export const InputsStatusUnion1$outboundSchema = z.union([
    InputsStatusCompleted1$outboundSchema,
    InputsStatusIncomplete1$outboundSchema,
    InputsStatusInProgress1$outboundSchema,
]);
export function inputsStatusUnion1ToJSON(inputsStatusUnion1) {
    return JSON.stringify(InputsStatusUnion1$outboundSchema.parse(inputsStatusUnion1));
}
/** @internal */
export const InputsContent1$outboundSchema = z.union([
    ResponseOutputText$outboundSchema,
    OpenAIResponsesRefusalContent$outboundSchema,
]);
export function inputsContent1ToJSON(inputsContent1) {
    return JSON.stringify(InputsContent1$outboundSchema.parse(inputsContent1));
}
/** @internal */
export const InputsContent2$outboundSchema = z.union([
    z.array(z.union([
        ResponseOutputText$outboundSchema,
        OpenAIResponsesRefusalContent$outboundSchema,
    ])),
    z.string(),
    z.any(),
]);
export function inputsContent2ToJSON(inputsContent2) {
    return JSON.stringify(InputsContent2$outboundSchema.parse(inputsContent2));
}
/** @internal */
export const InputsPhaseFinalAnswer$outboundSchema = z.enum(InputsPhaseFinalAnswer);
/** @internal */
export const InputsPhaseCommentary$outboundSchema = z.enum(InputsPhaseCommentary);
/** @internal */
export const InputsPhaseUnion$outboundSchema = z.union([
    InputsPhaseCommentary$outboundSchema,
    InputsPhaseFinalAnswer$outboundSchema,
    z.any(),
]);
export function inputsPhaseUnionToJSON(inputsPhaseUnion) {
    return JSON.stringify(InputsPhaseUnion$outboundSchema.parse(inputsPhaseUnion));
}
/** @internal */
export const InputsMessage$outboundSchema = z.object({
    id: z.string(),
    role: InputsRole$outboundSchema,
    type: InputsTypeMessage$outboundSchema,
    status: z.union([
        InputsStatusCompleted1$outboundSchema,
        InputsStatusIncomplete1$outboundSchema,
        InputsStatusInProgress1$outboundSchema,
    ]).optional(),
    content: z.nullable(z.union([
        z.array(z.union([
            ResponseOutputText$outboundSchema,
            OpenAIResponsesRefusalContent$outboundSchema,
        ])),
        z.string(),
        z.any(),
    ])),
    phase: z.nullable(z.union([
        InputsPhaseCommentary$outboundSchema,
        InputsPhaseFinalAnswer$outboundSchema,
        z.any(),
    ])).optional(),
});
export function inputsMessageToJSON(inputsMessage) {
    return JSON.stringify(InputsMessage$outboundSchema.parse(inputsMessage));
}
/** @internal */
export const InputsUnion1$outboundSchema = z.union([
    FunctionCallItem$outboundSchema,
    z.lazy(() => InputsMessage$outboundSchema),
    OutputFunctionCallItem$outboundSchema,
    OutputWebSearchCallItem$outboundSchema,
    OutputFileSearchCallItem$outboundSchema,
    OutputDatetimeItem$outboundSchema,
    ReasoningItem$outboundSchema,
    FunctionCallOutputItem$outboundSchema,
    z.lazy(() => InputsReasoning$outboundSchema),
    OutputImageGenerationCallItem$outboundSchema,
    OutputServerToolItem$outboundSchema,
    EasyInputMessage$outboundSchema,
    InputMessageItem$outboundSchema,
]);
export function inputsUnion1ToJSON(inputsUnion1) {
    return JSON.stringify(InputsUnion1$outboundSchema.parse(inputsUnion1));
}
/** @internal */
export const InputsUnion$outboundSchema = z.union([
    z.string(),
    z.array(z.union([
        FunctionCallItem$outboundSchema,
        z.lazy(() => InputsMessage$outboundSchema),
        OutputFunctionCallItem$outboundSchema,
        OutputWebSearchCallItem$outboundSchema,
        OutputFileSearchCallItem$outboundSchema,
        OutputDatetimeItem$outboundSchema,
        ReasoningItem$outboundSchema,
        FunctionCallOutputItem$outboundSchema,
        z.lazy(() => InputsReasoning$outboundSchema),
        OutputImageGenerationCallItem$outboundSchema,
        OutputServerToolItem$outboundSchema,
        EasyInputMessage$outboundSchema,
        InputMessageItem$outboundSchema,
    ])),
]);
export function inputsUnionToJSON(inputsUnion) {
    return JSON.stringify(InputsUnion$outboundSchema.parse(inputsUnion));
}
