// @openrouter/sdk/esm/models/inputfile.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 21319fb604ac
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const InputFile$inboundSchema = z.object({
    type: z.literal("input_file"),
    file_id: z.nullable(z.string()).optional(),
    file_data: z.string().optional(),
    filename: z.string().optional(),
    file_url: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        "file_id": "fileId",
        "file_data": "fileData",
        "file_url": "fileUrl",
    });
});
/** @internal */
export const InputFile$outboundSchema = z.object({
    type: z.literal("input_file"),
    fileId: z.nullable(z.string()).optional(),
    fileData: z.string().optional(),
    filename: z.string().optional(),
    fileUrl: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        fileId: "file_id",
        fileData: "file_data",
        fileUrl: "file_url",
    });
});
export function inputFileToJSON(inputFile) {
    return JSON.stringify(InputFile$outboundSchema.parse(inputFile));
}
export function inputFileFromJSON(jsonString) {
    return safeParse(jsonString, (x) => InputFile$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InputFile' from JSON`);
}
