// @openrouter/sdk/esm/models/inputmessageitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 78f212025019
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { InputAudio$outboundSchema, } from "./inputaudio.js";
import { InputFile$outboundSchema, } from "./inputfile.js";
import { InputText$outboundSchema, } from "./inputtext.js";
import { InputVideo$outboundSchema, } from "./inputvideo.js";
export const InputMessageItemTypeMessage = {
    Message: "message",
};
export const InputMessageItemRoleDeveloper = {
    Developer: "developer",
};
export const InputMessageItemRoleSystem = {
    System: "system",
};
export const InputMessageItemRoleUser = {
    User: "user",
};
export const InputMessageItemDetail = {
    Auto: "auto",
    High: "high",
    Low: "low",
};
/** @internal */
export const InputMessageItemTypeMessage$outboundSchema = z.enum(InputMessageItemTypeMessage);
/** @internal */
export const InputMessageItemRoleDeveloper$outboundSchema = z.enum(InputMessageItemRoleDeveloper);
/** @internal */
export const InputMessageItemRoleSystem$outboundSchema = z.enum(InputMessageItemRoleSystem);
/** @internal */
export const InputMessageItemRoleUser$outboundSchema = z.enum(InputMessageItemRoleUser);
/** @internal */
export const InputMessageItemRoleUnion$outboundSchema = z.union([
    InputMessageItemRoleUser$outboundSchema,
    InputMessageItemRoleSystem$outboundSchema,
    InputMessageItemRoleDeveloper$outboundSchema,
]);
export function inputMessageItemRoleUnionToJSON(inputMessageItemRoleUnion) {
    return JSON.stringify(InputMessageItemRoleUnion$outboundSchema.parse(inputMessageItemRoleUnion));
}
/** @internal */
export const InputMessageItemDetail$outboundSchema = openEnums.outboundSchema(InputMessageItemDetail);
/** @internal */
export const InputMessageItemContentInputImage$outboundSchema = z.object({
    type: z.literal("input_image"),
    detail: InputMessageItemDetail$outboundSchema,
    imageUrl: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        imageUrl: "image_url",
    });
});
export function inputMessageItemContentInputImageToJSON(inputMessageItemContentInputImage) {
    return JSON.stringify(InputMessageItemContentInputImage$outboundSchema.parse(inputMessageItemContentInputImage));
}
/** @internal */
export const InputMessageItemContentUnion$outboundSchema = z.union([
    InputText$outboundSchema,
    z.lazy(() => InputMessageItemContentInputImage$outboundSchema),
    InputFile$outboundSchema,
    InputAudio$outboundSchema,
    InputVideo$outboundSchema,
]);
export function inputMessageItemContentUnionToJSON(inputMessageItemContentUnion) {
    return JSON.stringify(InputMessageItemContentUnion$outboundSchema.parse(inputMessageItemContentUnion));
}
/** @internal */
export const InputMessageItem$outboundSchema = z.object({
    id: z.string().optional(),
    type: InputMessageItemTypeMessage$outboundSchema.optional(),
    role: z.union([
        InputMessageItemRoleUser$outboundSchema,
        InputMessageItemRoleSystem$outboundSchema,
        InputMessageItemRoleDeveloper$outboundSchema,
    ]),
    content: z.nullable(z.array(z.union([
        InputText$outboundSchema,
        z.lazy(() => InputMessageItemContentInputImage$outboundSchema),
        InputFile$outboundSchema,
        InputAudio$outboundSchema,
        InputVideo$outboundSchema,
    ]))).optional(),
});
export function inputMessageItemToJSON(inputMessageItem) {
    return JSON.stringify(InputMessageItem$outboundSchema.parse(inputMessageItem));
}
