// @openrouter/sdk/esm/models/legacywebsearchservertool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 1a9c64204177
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
import { SearchContextSizeEnum$inboundSchema, SearchContextSizeEnum$outboundSchema, } from "./searchcontextsizeenum.js";
import { WebSearchUserLocation$inboundSchema, WebSearchUserLocation$outboundSchema, } from "./websearchuserlocation.js";
/**
 * Which search engine to use. "auto" (default) uses native if the provider supports it, otherwise Exa. "native" forces the provider's built-in search. "exa" forces the Exa search API. "firecrawl" uses Firecrawl (requires BYOK). "parallel" uses the Parallel search API.
 */
export const LegacyWebSearchServerToolEngine = {
    Auto: "auto",
    Native: "native",
    Exa: "exa",
    Firecrawl: "firecrawl",
    Parallel: "parallel",
};
/** @internal */
export const LegacyWebSearchServerToolFilters$inboundSchema = z.object({
    allowed_domains: z.nullable(z.array(z.string())).optional(),
    excluded_domains: z.nullable(z.array(z.string())).optional(),
}).transform((v) => {
    return remap$(v, {
        "allowed_domains": "allowedDomains",
        "excluded_domains": "excludedDomains",
    });
});
/** @internal */
export const LegacyWebSearchServerToolFilters$outboundSchema = z.object({
    allowedDomains: z.nullable(z.array(z.string())).optional(),
    excludedDomains: z.nullable(z.array(z.string())).optional(),
}).transform((v) => {
    return remap$(v, {
        allowedDomains: "allowed_domains",
        excludedDomains: "excluded_domains",
    });
});
export function legacyWebSearchServerToolFiltersToJSON(legacyWebSearchServerToolFilters) {
    return JSON.stringify(LegacyWebSearchServerToolFilters$outboundSchema.parse(legacyWebSearchServerToolFilters));
}
export function legacyWebSearchServerToolFiltersFromJSON(jsonString) {
    return safeParse(jsonString, (x) => LegacyWebSearchServerToolFilters$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LegacyWebSearchServerToolFilters' from JSON`);
}
/** @internal */
export const LegacyWebSearchServerToolEngine$inboundSchema = openEnums.inboundSchema(LegacyWebSearchServerToolEngine);
/** @internal */
export const LegacyWebSearchServerToolEngine$outboundSchema = openEnums.outboundSchema(LegacyWebSearchServerToolEngine);
/** @internal */
export const LegacyWebSearchServerTool$inboundSchema = z.object({
    type: z.literal("web_search"),
    filters: z.nullable(z.lazy(() => LegacyWebSearchServerToolFilters$inboundSchema)).optional(),
    search_context_size: SearchContextSizeEnum$inboundSchema.optional(),
    user_location: z.nullable(WebSearchUserLocation$inboundSchema).optional(),
    engine: LegacyWebSearchServerToolEngine$inboundSchema.optional(),
    max_results: z.number().optional(),
}).transform((v) => {
    return remap$(v, {
        "search_context_size": "searchContextSize",
        "user_location": "userLocation",
        "max_results": "maxResults",
    });
});
/** @internal */
export const LegacyWebSearchServerTool$outboundSchema = z.object({
    type: z.literal("web_search"),
    filters: z.nullable(z.lazy(() => LegacyWebSearchServerToolFilters$outboundSchema)).optional(),
    searchContextSize: SearchContextSizeEnum$outboundSchema.optional(),
    userLocation: z.nullable(WebSearchUserLocation$outboundSchema).optional(),
    engine: LegacyWebSearchServerToolEngine$outboundSchema.optional(),
    maxResults: z.number().optional(),
}).transform((v) => {
    return remap$(v, {
        searchContextSize: "search_context_size",
        userLocation: "user_location",
        maxResults: "max_results",
    });
});
export function legacyWebSearchServerToolToJSON(legacyWebSearchServerTool) {
    return JSON.stringify(LegacyWebSearchServerTool$outboundSchema.parse(legacyWebSearchServerTool));
}
export function legacyWebSearchServerToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => LegacyWebSearchServerTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LegacyWebSearchServerTool' from JSON`);
}
