// @openrouter/sdk/esm/models/openresponsesresult.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 95391b93fcbf
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { ApplyPatchServerTool$inboundSchema, } from "./applypatchservertool.js";
import { BaseInputsUnion$inboundSchema, } from "./baseinputsunion.js";
import { BaseReasoningConfig$inboundSchema, } from "./basereasoningconfig.js";
import { CodeInterpreterServerTool$inboundSchema, } from "./codeinterpreterservertool.js";
import { CodexLocalShellTool$inboundSchema, } from "./codexlocalshelltool.js";
import { ComputerUseServerTool$inboundSchema, } from "./computeruseservertool.js";
import { CustomTool$inboundSchema } from "./customtool.js";
import { FileSearchServerTool$inboundSchema, } from "./filesearchservertool.js";
import { ImageGenerationServerTool$inboundSchema, } from "./imagegenerationservertool.js";
import { IncompleteDetails$inboundSchema, } from "./incompletedetails.js";
import { LegacyWebSearchServerTool$inboundSchema, } from "./legacywebsearchservertool.js";
import { McpServerTool$inboundSchema } from "./mcpservertool.js";
import { OpenAIResponsesResponseStatus$inboundSchema, } from "./openairesponsesresponsestatus.js";
import { OpenAIResponsesToolChoiceUnion$inboundSchema, } from "./openairesponsestoolchoiceunion.js";
import { OutputItems$inboundSchema } from "./outputitems.js";
import { Preview20250311WebSearchServerTool$inboundSchema, } from "./preview20250311websearchservertool.js";
import { PreviewWebSearchServerTool$inboundSchema, } from "./previewwebsearchservertool.js";
import { ResponsesErrorField$inboundSchema, } from "./responseserrorfield.js";
import { ShellServerTool$inboundSchema, } from "./shellservertool.js";
import { StoredPromptTemplate$inboundSchema, } from "./storedprompttemplate.js";
import { TextConfig$inboundSchema } from "./textconfig.js";
import { TruncationEnum$inboundSchema, } from "./truncationenum.js";
import { Usage$inboundSchema } from "./usage.js";
import { WebSearchServerTool$inboundSchema, } from "./websearchservertool.js";
export const OpenResponsesResultObject = {
    Response: "response",
};
/** @internal */
export const OpenResponsesResultObject$inboundSchema = z.enum(OpenResponsesResultObject);
/** @internal */
export const OpenResponsesResultToolFunction$inboundSchema = z.object({
    type: z.literal("function"),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    strict: z.nullable(z.boolean()).optional(),
    parameters: z.nullable(z.record(z.string(), z.nullable(z.any()))),
});
export function openResponsesResultToolFunctionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesResultToolFunction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesResultToolFunction' from JSON`);
}
/** @internal */
export const OpenResponsesResultToolUnion$inboundSchema = z.union([
    z.lazy(() => OpenResponsesResultToolFunction$inboundSchema),
    PreviewWebSearchServerTool$inboundSchema,
    Preview20250311WebSearchServerTool$inboundSchema,
    LegacyWebSearchServerTool$inboundSchema,
    WebSearchServerTool$inboundSchema,
    FileSearchServerTool$inboundSchema,
    ComputerUseServerTool$inboundSchema,
    CodeInterpreterServerTool$inboundSchema,
    McpServerTool$inboundSchema,
    ImageGenerationServerTool$inboundSchema,
    CodexLocalShellTool$inboundSchema,
    ShellServerTool$inboundSchema,
    ApplyPatchServerTool$inboundSchema,
    CustomTool$inboundSchema,
]);
export function openResponsesResultToolUnionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesResultToolUnion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesResultToolUnion' from JSON`);
}
/** @internal */
export const OpenResponsesResult$inboundSchema = z.object({
    id: z.string(),
    object: OpenResponsesResultObject$inboundSchema,
    created_at: z.number(),
    model: z.string(),
    status: OpenAIResponsesResponseStatus$inboundSchema,
    completed_at: z.nullable(z.number()),
    output: z.array(OutputItems$inboundSchema),
    user: z.nullable(z.string()).optional(),
    output_text: z.string().optional(),
    prompt_cache_key: z.nullable(z.string()).optional(),
    safety_identifier: z.nullable(z.string()).optional(),
    error: z.nullable(ResponsesErrorField$inboundSchema),
    incomplete_details: z.nullable(IncompleteDetails$inboundSchema),
    usage: z.nullable(Usage$inboundSchema).optional(),
    max_tool_calls: z.nullable(z.number()).optional(),
    top_logprobs: z.number().optional(),
    max_output_tokens: z.nullable(z.number()).optional(),
    temperature: z.nullable(z.number()),
    top_p: z.nullable(z.number()),
    presence_penalty: z.nullable(z.number()),
    frequency_penalty: z.nullable(z.number()),
    instructions: z.nullable(BaseInputsUnion$inboundSchema),
    metadata: z.nullable(z.record(z.string(), z.string())),
    tools: z.array(z.union([
        z.lazy(() => OpenResponsesResultToolFunction$inboundSchema),
        PreviewWebSearchServerTool$inboundSchema,
        Preview20250311WebSearchServerTool$inboundSchema,
        LegacyWebSearchServerTool$inboundSchema,
        WebSearchServerTool$inboundSchema,
        FileSearchServerTool$inboundSchema,
        ComputerUseServerTool$inboundSchema,
        CodeInterpreterServerTool$inboundSchema,
        McpServerTool$inboundSchema,
        ImageGenerationServerTool$inboundSchema,
        CodexLocalShellTool$inboundSchema,
        ShellServerTool$inboundSchema,
        ApplyPatchServerTool$inboundSchema,
        CustomTool$inboundSchema,
    ])),
    tool_choice: OpenAIResponsesToolChoiceUnion$inboundSchema,
    parallel_tool_calls: z.boolean(),
    prompt: z.nullable(StoredPromptTemplate$inboundSchema).optional(),
    background: z.nullable(z.boolean()).optional(),
    previous_response_id: z.nullable(z.string()).optional(),
    reasoning: z.nullable(BaseReasoningConfig$inboundSchema).optional(),
    service_tier: z.nullable(z.string()).optional(),
    store: z.boolean().optional(),
    truncation: z.nullable(TruncationEnum$inboundSchema).optional(),
    text: TextConfig$inboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "completed_at": "completedAt",
        "output_text": "outputText",
        "prompt_cache_key": "promptCacheKey",
        "safety_identifier": "safetyIdentifier",
        "incomplete_details": "incompleteDetails",
        "max_tool_calls": "maxToolCalls",
        "top_logprobs": "topLogprobs",
        "max_output_tokens": "maxOutputTokens",
        "top_p": "topP",
        "presence_penalty": "presencePenalty",
        "frequency_penalty": "frequencyPenalty",
        "tool_choice": "toolChoice",
        "parallel_tool_calls": "parallelToolCalls",
        "previous_response_id": "previousResponseId",
        "service_tier": "serviceTier",
    });
});
export function openResponsesResultFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OpenResponsesResult$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OpenResponsesResult' from JSON`);
}
