// @openrouter/sdk/esm/models/outputreasoningitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 4ea30f8ad34c
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
import { ReasoningSummaryText$inboundSchema, } from "./reasoningsummarytext.js";
import { ReasoningTextContent$inboundSchema, } from "./reasoningtextcontent.js";
export const OutputReasoningItemType = {
    Reasoning: "reasoning",
};
export const OutputReasoningItemStatusInProgress = {
    InProgress: "in_progress",
};
export const OutputReasoningItemStatusIncomplete = {
    Incomplete: "incomplete",
};
export const OutputReasoningItemStatusCompleted = {
    Completed: "completed",
};
/**
 * The format of the reasoning content
 */
export const OutputReasoningItemFormat = {
    Unknown: "unknown",
    OpenaiResponsesV1: "openai-responses-v1",
    AzureOpenaiResponsesV1: "azure-openai-responses-v1",
    XaiResponsesV1: "xai-responses-v1",
    AnthropicClaudeV1: "anthropic-claude-v1",
    GoogleGeminiV1: "google-gemini-v1",
};
/** @internal */
export const OutputReasoningItemType$inboundSchema = z.enum(OutputReasoningItemType);
/** @internal */
export const OutputReasoningItemStatusInProgress$inboundSchema = z.enum(OutputReasoningItemStatusInProgress);
/** @internal */
export const OutputReasoningItemStatusIncomplete$inboundSchema = z.enum(OutputReasoningItemStatusIncomplete);
/** @internal */
export const OutputReasoningItemStatusCompleted$inboundSchema = z.enum(OutputReasoningItemStatusCompleted);
/** @internal */
export const OutputReasoningItemStatusUnion$inboundSchema = z.union([
    OutputReasoningItemStatusCompleted$inboundSchema,
    OutputReasoningItemStatusIncomplete$inboundSchema,
    OutputReasoningItemStatusInProgress$inboundSchema,
]);
export function outputReasoningItemStatusUnionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputReasoningItemStatusUnion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputReasoningItemStatusUnion' from JSON`);
}
/** @internal */
export const OutputReasoningItemFormat$inboundSchema = openEnums.inboundSchema(OutputReasoningItemFormat);
/** @internal */
export const OutputReasoningItem$inboundSchema = z.object({
    type: OutputReasoningItemType$inboundSchema,
    id: z.string(),
    content: z.nullable(z.array(ReasoningTextContent$inboundSchema)).optional(),
    summary: z.array(ReasoningSummaryText$inboundSchema),
    encrypted_content: z.nullable(z.string()).optional(),
    status: z.union([
        OutputReasoningItemStatusCompleted$inboundSchema,
        OutputReasoningItemStatusIncomplete$inboundSchema,
        OutputReasoningItemStatusInProgress$inboundSchema,
    ]).optional(),
    signature: z.nullable(z.string()).optional(),
    format: z.nullable(OutputReasoningItemFormat$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "encrypted_content": "encryptedContent",
    });
});
export function outputReasoningItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputReasoningItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputReasoningItem' from JSON`);
}
