// @openrouter/sdk/esm/models/outputservertoolitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: ca93d8ce3ebb
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { collectExtraKeys as collectExtraKeys$, safeParse, } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
export const OutputServerToolItemStatus = {
    Completed: "completed",
    InProgress: "in_progress",
    Incomplete: "incomplete",
};
/** @internal */
export const OutputServerToolItemStatus$inboundSchema = openEnums.inboundSchema(OutputServerToolItemStatus);
/** @internal */
export const OutputServerToolItemStatus$outboundSchema = openEnums.outboundSchema(OutputServerToolItemStatus);
/** @internal */
export const OutputServerToolItem$inboundSchema = collectExtraKeys$(z.object({
    type: z.string(),
    id: z.string().optional(),
    status: OutputServerToolItemStatus$inboundSchema,
}).catchall(z.any()), "additionalProperties", true);
/** @internal */
export const OutputServerToolItem$outboundSchema = z.object({
    type: z.string(),
    id: z.string().optional(),
    status: OutputServerToolItemStatus$outboundSchema,
    additionalProperties: z.record(z.string(), z.nullable(z.any())).optional(),
}).transform((v) => {
    return {
        ...v.additionalProperties,
        ...remap$(v, {
            additionalProperties: null,
        }),
    };
});
export function outputServerToolItemToJSON(outputServerToolItem) {
    return JSON.stringify(OutputServerToolItem$outboundSchema.parse(outputServerToolItem));
}
export function outputServerToolItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputServerToolItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputServerToolItem' from JSON`);
}
