// @openrouter/sdk/esm/models/outputmessageitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 7eb39195329c
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
import { OpenAIResponsesRefusalContent$inboundSchema, } from "./openairesponsesrefusalcontent.js";
import { ResponseOutputText$inboundSchema, } from "./responseoutputtext.js";
export const OutputMessageItemRole = {
    Assistant: "assistant",
};
export const OutputMessageItemType = {
    Message: "message",
};
export const OutputMessageItemStatusInProgress = {
    InProgress: "in_progress",
};
export const OutputMessageItemStatusIncomplete = {
    Incomplete: "incomplete",
};
export const OutputMessageItemStatusCompleted = {
    Completed: "completed",
};
export const OutputMessageItemPhaseFinalAnswer = {
    FinalAnswer: "final_answer",
};
export const OutputMessageItemPhaseCommentary = {
    Commentary: "commentary",
};
/** @internal */
export const OutputMessageItemRole$inboundSchema = z.enum(OutputMessageItemRole);
/** @internal */
export const OutputMessageItemType$inboundSchema = z.enum(OutputMessageItemType);
/** @internal */
export const OutputMessageItemStatusInProgress$inboundSchema = z.enum(OutputMessageItemStatusInProgress);
/** @internal */
export const OutputMessageItemStatusIncomplete$inboundSchema = z.enum(OutputMessageItemStatusIncomplete);
/** @internal */
export const OutputMessageItemStatusCompleted$inboundSchema = z.enum(OutputMessageItemStatusCompleted);
/** @internal */
export const OutputMessageItemStatusUnion$inboundSchema = z.union([
    OutputMessageItemStatusCompleted$inboundSchema,
    OutputMessageItemStatusIncomplete$inboundSchema,
    OutputMessageItemStatusInProgress$inboundSchema,
]);
export function outputMessageItemStatusUnionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputMessageItemStatusUnion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputMessageItemStatusUnion' from JSON`);
}
/** @internal */
export const OutputMessageItemContent$inboundSchema = z.union([
    ResponseOutputText$inboundSchema,
    OpenAIResponsesRefusalContent$inboundSchema,
]);
export function outputMessageItemContentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputMessageItemContent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputMessageItemContent' from JSON`);
}
/** @internal */
export const OutputMessageItemPhaseFinalAnswer$inboundSchema = z.enum(OutputMessageItemPhaseFinalAnswer);
/** @internal */
export const OutputMessageItemPhaseCommentary$inboundSchema = z.enum(OutputMessageItemPhaseCommentary);
/** @internal */
export const OutputMessageItemPhaseUnion$inboundSchema = z.union([
    OutputMessageItemPhaseCommentary$inboundSchema,
    OutputMessageItemPhaseFinalAnswer$inboundSchema,
    z.any(),
]);
export function outputMessageItemPhaseUnionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputMessageItemPhaseUnion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputMessageItemPhaseUnion' from JSON`);
}
/** @internal */
export const OutputMessageItem$inboundSchema = z.object({
    id: z.string(),
    role: OutputMessageItemRole$inboundSchema,
    type: OutputMessageItemType$inboundSchema,
    status: z.union([
        OutputMessageItemStatusCompleted$inboundSchema,
        OutputMessageItemStatusIncomplete$inboundSchema,
        OutputMessageItemStatusInProgress$inboundSchema,
    ]).optional(),
    content: z.array(z.union([
        ResponseOutputText$inboundSchema,
        OpenAIResponsesRefusalContent$inboundSchema,
    ])),
    phase: z.nullable(z.union([
        OutputMessageItemPhaseCommentary$inboundSchema,
        OutputMessageItemPhaseFinalAnswer$inboundSchema,
        z.any(),
    ])).optional(),
});
export function outputMessageItemFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OutputMessageItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OutputMessageItem' from JSON`);
}
