// @openrouter/sdk/esm/models/outputdatetimeitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 0c1828e72f98
 */
import * as z from "zod/v4";
import * as openEnums from "../types/enums.js";
export const OutputDatetimeItemType = {
    OpenrouterDatetime: "openrouter:datetime",
};
export const OutputDatetimeItemStatus = {
    Completed: "completed",
    InProgress: "in_progress",
    Incomplete: "incomplete",
};
/** @internal */
export const OutputDatetimeItemType$outboundSchema = z.enum(OutputDatetimeItemType);
/** @internal */
export const OutputDatetimeItemStatus$outboundSchema = openEnums.outboundSchema(OutputDatetimeItemStatus);
/** @internal */
export const OutputDatetimeItem$outboundSchema = z.object({
    type: OutputDatetimeItemType$outboundSchema,
    id: z.string().optional(),
    status: OutputDatetimeItemStatus$outboundSchema,
    datetime: z.string(),
    timezone: z.string(),
});
export function outputDatetimeItemToJSON(outputDatetimeItem) {
    return JSON.stringify(OutputDatetimeItem$outboundSchema.parse(outputDatetimeItem));
}
