// @openrouter/sdk/esm/models/preview20250311websearchservertool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: c171c186ccac
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
import { PreviewWebSearchUserLocation$inboundSchema, PreviewWebSearchUserLocation$outboundSchema, } from "./previewwebsearchuserlocation.js";
import { SearchContextSizeEnum$inboundSchema, SearchContextSizeEnum$outboundSchema, } from "./searchcontextsizeenum.js";
/**
 * Which search engine to use. "auto" (default) uses native if the provider supports it, otherwise Exa. "native" forces the provider's built-in search. "exa" forces the Exa search API. "firecrawl" uses Firecrawl (requires BYOK). "parallel" uses the Parallel search API.
 */
export const Preview20250311WebSearchServerToolEngine = {
    Auto: "auto",
    Native: "native",
    Exa: "exa",
    Firecrawl: "firecrawl",
    Parallel: "parallel",
};
/** @internal */
export const Preview20250311WebSearchServerToolEngine$inboundSchema = openEnums.inboundSchema(Preview20250311WebSearchServerToolEngine);
/** @internal */
export const Preview20250311WebSearchServerToolEngine$outboundSchema = openEnums.outboundSchema(Preview20250311WebSearchServerToolEngine);
/** @internal */
export const Preview20250311WebSearchServerToolFilters$inboundSchema = z.object({
    allowed_domains: z.nullable(z.array(z.string())).optional(),
    excluded_domains: z.nullable(z.array(z.string())).optional(),
}).transform((v) => {
    return remap$(v, {
        "allowed_domains": "allowedDomains",
        "excluded_domains": "excludedDomains",
    });
});
/** @internal */
export const Preview20250311WebSearchServerToolFilters$outboundSchema = z.object({
    allowedDomains: z.nullable(z.array(z.string())).optional(),
    excludedDomains: z.nullable(z.array(z.string())).optional(),
}).transform((v) => {
    return remap$(v, {
        allowedDomains: "allowed_domains",
        excludedDomains: "excluded_domains",
    });
});
export function preview20250311WebSearchServerToolFiltersToJSON(preview20250311WebSearchServerToolFilters) {
    return JSON.stringify(Preview20250311WebSearchServerToolFilters$outboundSchema.parse(preview20250311WebSearchServerToolFilters));
}
export function preview20250311WebSearchServerToolFiltersFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Preview20250311WebSearchServerToolFilters$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Preview20250311WebSearchServerToolFilters' from JSON`);
}
/** @internal */
export const Preview20250311WebSearchServerTool$inboundSchema = z.object({
    type: z.literal("web_search_preview_2025_03_11"),
    search_context_size: SearchContextSizeEnum$inboundSchema.optional(),
    user_location: z.nullable(PreviewWebSearchUserLocation$inboundSchema)
        .optional(),
    engine: Preview20250311WebSearchServerToolEngine$inboundSchema.optional(),
    max_results: z.number().optional(),
    filters: z.nullable(z.lazy(() => Preview20250311WebSearchServerToolFilters$inboundSchema)).optional(),
}).transform((v) => {
    return remap$(v, {
        "search_context_size": "searchContextSize",
        "user_location": "userLocation",
        "max_results": "maxResults",
    });
});
/** @internal */
export const Preview20250311WebSearchServerTool$outboundSchema = z.object({
    type: z.literal("web_search_preview_2025_03_11"),
    searchContextSize: SearchContextSizeEnum$outboundSchema.optional(),
    userLocation: z.nullable(PreviewWebSearchUserLocation$outboundSchema)
        .optional(),
    engine: Preview20250311WebSearchServerToolEngine$outboundSchema.optional(),
    maxResults: z.number().optional(),
    filters: z.nullable(z.lazy(() => Preview20250311WebSearchServerToolFilters$outboundSchema)).optional(),
}).transform((v) => {
    return remap$(v, {
        searchContextSize: "search_context_size",
        userLocation: "user_location",
        maxResults: "max_results",
    });
});
export function preview20250311WebSearchServerToolToJSON(preview20250311WebSearchServerTool) {
    return JSON.stringify(Preview20250311WebSearchServerTool$outboundSchema.parse(preview20250311WebSearchServerTool));
}
export function preview20250311WebSearchServerToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Preview20250311WebSearchServerTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Preview20250311WebSearchServerTool' from JSON`);
}
