// @openrouter/sdk/esm/models/previewwebsearchservertool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 6b7147491ff2
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
import { PreviewWebSearchUserLocation$inboundSchema, PreviewWebSearchUserLocation$outboundSchema, } from "./previewwebsearchuserlocation.js";
import { SearchContextSizeEnum$inboundSchema, SearchContextSizeEnum$outboundSchema, } from "./searchcontextsizeenum.js";
/**
 * Which search engine to use. "auto" (default) uses native if the provider supports it, otherwise Exa. "native" forces the provider's built-in search. "exa" forces the Exa search API. "firecrawl" uses Firecrawl (requires BYOK). "parallel" uses the Parallel search API.
 */
export const PreviewWebSearchServerToolEngine = {
    Auto: "auto",
    Native: "native",
    Exa: "exa",
    Firecrawl: "firecrawl",
    Parallel: "parallel",
};
/** @internal */
export const PreviewWebSearchServerToolEngine$inboundSchema = openEnums.inboundSchema(PreviewWebSearchServerToolEngine);
/** @internal */
export const PreviewWebSearchServerToolEngine$outboundSchema = openEnums.outboundSchema(PreviewWebSearchServerToolEngine);
/** @internal */
export const PreviewWebSearchServerToolFilters$inboundSchema = z.object({
    allowed_domains: z.nullable(z.array(z.string())).optional(),
    excluded_domains: z.nullable(z.array(z.string())).optional(),
}).transform((v) => {
    return remap$(v, {
        "allowed_domains": "allowedDomains",
        "excluded_domains": "excludedDomains",
    });
});
/** @internal */
export const PreviewWebSearchServerToolFilters$outboundSchema = z.object({
    allowedDomains: z.nullable(z.array(z.string())).optional(),
    excludedDomains: z.nullable(z.array(z.string())).optional(),
}).transform((v) => {
    return remap$(v, {
        allowedDomains: "allowed_domains",
        excludedDomains: "excluded_domains",
    });
});
export function previewWebSearchServerToolFiltersToJSON(previewWebSearchServerToolFilters) {
    return JSON.stringify(PreviewWebSearchServerToolFilters$outboundSchema.parse(previewWebSearchServerToolFilters));
}
export function previewWebSearchServerToolFiltersFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PreviewWebSearchServerToolFilters$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PreviewWebSearchServerToolFilters' from JSON`);
}
/** @internal */
export const PreviewWebSearchServerTool$inboundSchema = z.object({
    type: z.literal("web_search_preview"),
    search_context_size: SearchContextSizeEnum$inboundSchema.optional(),
    user_location: z.nullable(PreviewWebSearchUserLocation$inboundSchema)
        .optional(),
    engine: PreviewWebSearchServerToolEngine$inboundSchema.optional(),
    max_results: z.number().optional(),
    filters: z.nullable(z.lazy(() => PreviewWebSearchServerToolFilters$inboundSchema)).optional(),
}).transform((v) => {
    return remap$(v, {
        "search_context_size": "searchContextSize",
        "user_location": "userLocation",
        "max_results": "maxResults",
    });
});
/** @internal */
export const PreviewWebSearchServerTool$outboundSchema = z.object({
    type: z.literal("web_search_preview"),
    searchContextSize: SearchContextSizeEnum$outboundSchema.optional(),
    userLocation: z.nullable(PreviewWebSearchUserLocation$outboundSchema)
        .optional(),
    engine: PreviewWebSearchServerToolEngine$outboundSchema.optional(),
    maxResults: z.number().optional(),
    filters: z.nullable(z.lazy(() => PreviewWebSearchServerToolFilters$outboundSchema)).optional(),
}).transform((v) => {
    return remap$(v, {
        searchContextSize: "search_context_size",
        userLocation: "user_location",
        maxResults: "max_results",
    });
});
export function previewWebSearchServerToolToJSON(previewWebSearchServerTool) {
    return JSON.stringify(PreviewWebSearchServerTool$outboundSchema.parse(previewWebSearchServerTool));
}
export function previewWebSearchServerToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PreviewWebSearchServerTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PreviewWebSearchServerTool' from JSON`);
}
