// @openrouter/sdk/esm/models/reasoningitem.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 347a000c48b1
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { ReasoningSummaryText$outboundSchema, } from "./reasoningsummarytext.js";
import { ReasoningTextContent$outboundSchema, } from "./reasoningtextcontent.js";
export const ReasoningItemType = {
    Reasoning: "reasoning",
};
export const ReasoningItemStatusInProgress = {
    InProgress: "in_progress",
};
export const ReasoningItemStatusIncomplete = {
    Incomplete: "incomplete",
};
export const ReasoningItemStatusCompleted = {
    Completed: "completed",
};
export const ReasoningItemFormat = {
    Unknown: "unknown",
    OpenaiResponsesV1: "openai-responses-v1",
    AzureOpenaiResponsesV1: "azure-openai-responses-v1",
    XaiResponsesV1: "xai-responses-v1",
    AnthropicClaudeV1: "anthropic-claude-v1",
    GoogleGeminiV1: "google-gemini-v1",
};
/** @internal */
export const ReasoningItemType$outboundSchema = z.enum(ReasoningItemType);
/** @internal */
export const ReasoningItemStatusInProgress$outboundSchema = z.enum(ReasoningItemStatusInProgress);
/** @internal */
export const ReasoningItemStatusIncomplete$outboundSchema = z.enum(ReasoningItemStatusIncomplete);
/** @internal */
export const ReasoningItemStatusCompleted$outboundSchema = z.enum(ReasoningItemStatusCompleted);
/** @internal */
export const ReasoningItemStatusUnion$outboundSchema = z.union([
    ReasoningItemStatusCompleted$outboundSchema,
    ReasoningItemStatusIncomplete$outboundSchema,
    ReasoningItemStatusInProgress$outboundSchema,
]);
export function reasoningItemStatusUnionToJSON(reasoningItemStatusUnion) {
    return JSON.stringify(ReasoningItemStatusUnion$outboundSchema.parse(reasoningItemStatusUnion));
}
/** @internal */
export const ReasoningItemFormat$outboundSchema = openEnums.outboundSchema(ReasoningItemFormat);
/** @internal */
export const ReasoningItem$outboundSchema = z.object({
    type: ReasoningItemType$outboundSchema,
    id: z.string(),
    content: z.nullable(z.array(ReasoningTextContent$outboundSchema)).optional(),
    summary: z.array(ReasoningSummaryText$outboundSchema),
    encryptedContent: z.nullable(z.string()).optional(),
    status: z.union([
        ReasoningItemStatusCompleted$outboundSchema,
        ReasoningItemStatusIncomplete$outboundSchema,
        ReasoningItemStatusInProgress$outboundSchema,
    ]).optional(),
    signature: z.nullable(z.string()).optional(),
    format: z.nullable(ReasoningItemFormat$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        encryptedContent: "encrypted_content",
    });
});
export function reasoningItemToJSON(reasoningItem) {
    return JSON.stringify(ReasoningItem$outboundSchema.parse(reasoningItem));
}
