// @openrouter/sdk/esm/models/responsesrequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 39799702f982
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import * as openEnums from "../types/enums.js";
import { ApplyPatchServerTool$outboundSchema, } from "./applypatchservertool.js";
import { CodeInterpreterServerTool$outboundSchema, } from "./codeinterpreterservertool.js";
import { CodexLocalShellTool$outboundSchema, } from "./codexlocalshelltool.js";
import { ComputerUseServerTool$outboundSchema, } from "./computeruseservertool.js";
import { ContextCompressionEngine$outboundSchema, } from "./contextcompressionengine.js";
import { CustomTool$outboundSchema, } from "./customtool.js";
import { DataCollection$outboundSchema, } from "./datacollection.js";
import { DatetimeServerTool$outboundSchema, } from "./datetimeservertool.js";
import { FileSearchServerTool$outboundSchema, } from "./filesearchservertool.js";
import { ImageGenerationServerTool$outboundSchema, } from "./imagegenerationservertool.js";
import { InputsUnion$outboundSchema, } from "./inputsunion.js";
import { LegacyWebSearchServerTool$outboundSchema, } from "./legacywebsearchservertool.js";
import { McpServerTool$outboundSchema, } from "./mcpservertool.js";
import { OpenAIResponsesToolChoiceUnion$outboundSchema, } from "./openairesponsestoolchoiceunion.js";
import { OpenAIResponsesTruncation$outboundSchema, } from "./openairesponsestruncation.js";
import { OutputModalityEnum$outboundSchema, } from "./outputmodalityenum.js";
import { PDFParserOptions$outboundSchema, } from "./pdfparseroptions.js";
import { PreferredMaxLatency$outboundSchema, } from "./preferredmaxlatency.js";
import { PreferredMinThroughput$outboundSchema, } from "./preferredminthroughput.js";
import { Preview20250311WebSearchServerTool$outboundSchema, } from "./preview20250311websearchservertool.js";
import { PreviewWebSearchServerTool$outboundSchema, } from "./previewwebsearchservertool.js";
import { ProviderName$outboundSchema } from "./providername.js";
import { ProviderSort$outboundSchema } from "./providersort.js";
import { ProviderSortConfig$outboundSchema, } from "./providersortconfig.js";
import { Quantization$outboundSchema } from "./quantization.js";
import { ReasoningConfig$outboundSchema, } from "./reasoningconfig.js";
import { ResponseIncludesEnum$outboundSchema, } from "./responseincludesenum.js";
import { ShellServerTool$outboundSchema, } from "./shellservertool.js";
import { StoredPromptTemplate$outboundSchema, } from "./storedprompttemplate.js";
import { TextExtendedConfig$outboundSchema, } from "./textextendedconfig.js";
import { WebSearchEngine$outboundSchema, } from "./websearchengine.js";
import { WebSearchServerTool$outboundSchema, } from "./websearchservertool.js";
import { WebSearchServerToolOpenRouter$outboundSchema, } from "./websearchservertoolopenrouter.js";
export const ResponsesRequestServiceTier = {
    Auto: "auto",
    Default: "default",
    Flex: "flex",
    Priority: "priority",
    Scale: "scale",
};
/** @internal */
export const ResponsesRequestToolFunction$outboundSchema = z.object({
    type: z.literal("function"),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    strict: z.nullable(z.boolean()).optional(),
    parameters: z.nullable(z.record(z.string(), z.nullable(z.any()))),
});
export function responsesRequestToolFunctionToJSON(responsesRequestToolFunction) {
    return JSON.stringify(ResponsesRequestToolFunction$outboundSchema.parse(responsesRequestToolFunction));
}
/** @internal */
export const ResponsesRequestToolUnion$outboundSchema = z.union([
    z.lazy(() => ResponsesRequestToolFunction$outboundSchema),
    PreviewWebSearchServerTool$outboundSchema,
    Preview20250311WebSearchServerTool$outboundSchema,
    LegacyWebSearchServerTool$outboundSchema,
    WebSearchServerTool$outboundSchema,
    FileSearchServerTool$outboundSchema,
    ComputerUseServerTool$outboundSchema,
    CodeInterpreterServerTool$outboundSchema,
    McpServerTool$outboundSchema,
    ImageGenerationServerTool$outboundSchema,
    CodexLocalShellTool$outboundSchema,
    ShellServerTool$outboundSchema,
    ApplyPatchServerTool$outboundSchema,
    CustomTool$outboundSchema,
    DatetimeServerTool$outboundSchema.and(z.object({ type: z.literal("openrouter:datetime") })),
    WebSearchServerToolOpenRouter$outboundSchema,
]);
export function responsesRequestToolUnionToJSON(responsesRequestToolUnion) {
    return JSON.stringify(ResponsesRequestToolUnion$outboundSchema.parse(responsesRequestToolUnion));
}
/** @internal */
export const ResponsesRequestImageConfig$outboundSchema = z.union([z.string(), z.number()]);
export function responsesRequestImageConfigToJSON(responsesRequestImageConfig) {
    return JSON.stringify(ResponsesRequestImageConfig$outboundSchema.parse(responsesRequestImageConfig));
}
/** @internal */
export const ResponsesRequestServiceTier$outboundSchema = openEnums.outboundSchema(ResponsesRequestServiceTier);
/** @internal */
export const ResponsesRequestOrder$outboundSchema = z.union([ProviderName$outboundSchema, z.string()]);
export function responsesRequestOrderToJSON(responsesRequestOrder) {
    return JSON.stringify(ResponsesRequestOrder$outboundSchema.parse(responsesRequestOrder));
}
/** @internal */
export const ResponsesRequestOnly$outboundSchema = z.union([ProviderName$outboundSchema, z.string()]);
export function responsesRequestOnlyToJSON(responsesRequestOnly) {
    return JSON.stringify(ResponsesRequestOnly$outboundSchema.parse(responsesRequestOnly));
}
/** @internal */
export const ResponsesRequestIgnore$outboundSchema = z.union([ProviderName$outboundSchema, z.string()]);
export function responsesRequestIgnoreToJSON(responsesRequestIgnore) {
    return JSON.stringify(ResponsesRequestIgnore$outboundSchema.parse(responsesRequestIgnore));
}
/** @internal */
export const ResponsesRequestSort$outboundSchema = z.union([
    ProviderSort$outboundSchema,
    ProviderSortConfig$outboundSchema,
    z.any(),
]);
export function responsesRequestSortToJSON(responsesRequestSort) {
    return JSON.stringify(ResponsesRequestSort$outboundSchema.parse(responsesRequestSort));
}
/** @internal */
export const ResponsesRequestMaxPrice$outboundSchema = z.object({
    prompt: z.string().optional(),
    completion: z.string().optional(),
    image: z.string().optional(),
    audio: z.string().optional(),
    request: z.string().optional(),
});
export function responsesRequestMaxPriceToJSON(responsesRequestMaxPrice) {
    return JSON.stringify(ResponsesRequestMaxPrice$outboundSchema.parse(responsesRequestMaxPrice));
}
/** @internal */
export const ResponsesRequestProvider$outboundSchema = z.object({
    allowFallbacks: z.nullable(z.boolean()).optional(),
    requireParameters: z.nullable(z.boolean()).optional(),
    dataCollection: z.nullable(DataCollection$outboundSchema).optional(),
    zdr: z.nullable(z.boolean()).optional(),
    enforceDistillableText: z.nullable(z.boolean()).optional(),
    order: z.nullable(z.array(z.union([ProviderName$outboundSchema, z.string()])))
        .optional(),
    only: z.nullable(z.array(z.union([ProviderName$outboundSchema, z.string()])))
        .optional(),
    ignore: z.nullable(z.array(z.union([ProviderName$outboundSchema, z.string()]))).optional(),
    quantizations: z.nullable(z.array(Quantization$outboundSchema)).optional(),
    sort: z.nullable(z.union([
        ProviderSort$outboundSchema,
        ProviderSortConfig$outboundSchema,
        z.any(),
    ])).optional(),
    maxPrice: z.lazy(() => ResponsesRequestMaxPrice$outboundSchema).optional(),
    preferredMinThroughput: z.nullable(PreferredMinThroughput$outboundSchema)
        .optional(),
    preferredMaxLatency: z.nullable(PreferredMaxLatency$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        allowFallbacks: "allow_fallbacks",
        requireParameters: "require_parameters",
        dataCollection: "data_collection",
        enforceDistillableText: "enforce_distillable_text",
        maxPrice: "max_price",
        preferredMinThroughput: "preferred_min_throughput",
        preferredMaxLatency: "preferred_max_latency",
    });
});
export function responsesRequestProviderToJSON(responsesRequestProvider) {
    return JSON.stringify(ResponsesRequestProvider$outboundSchema.parse(responsesRequestProvider));
}
/** @internal */
export const ResponsesRequestPluginContextCompression$outboundSchema = z.object({
    id: z.literal("context-compression"),
    enabled: z.boolean().optional(),
    engine: ContextCompressionEngine$outboundSchema.optional(),
});
export function responsesRequestPluginContextCompressionToJSON(responsesRequestPluginContextCompression) {
    return JSON.stringify(ResponsesRequestPluginContextCompression$outboundSchema.parse(responsesRequestPluginContextCompression));
}
/** @internal */
export const ResponsesRequestPluginResponseHealing$outboundSchema = z.object({
    id: z.literal("response-healing"),
    enabled: z.boolean().optional(),
});
export function responsesRequestPluginResponseHealingToJSON(responsesRequestPluginResponseHealing) {
    return JSON.stringify(ResponsesRequestPluginResponseHealing$outboundSchema.parse(responsesRequestPluginResponseHealing));
}
/** @internal */
export const ResponsesRequestPluginFileParser$outboundSchema = z.object({
    id: z.literal("file-parser"),
    enabled: z.boolean().optional(),
    pdf: PDFParserOptions$outboundSchema.optional(),
});
export function responsesRequestPluginFileParserToJSON(responsesRequestPluginFileParser) {
    return JSON.stringify(ResponsesRequestPluginFileParser$outboundSchema.parse(responsesRequestPluginFileParser));
}
/** @internal */
export const ResponsesRequestPluginWeb$outboundSchema = z.object({
    id: z.literal("web"),
    enabled: z.boolean().optional(),
    maxResults: z.number().optional(),
    searchPrompt: z.string().optional(),
    engine: WebSearchEngine$outboundSchema.optional(),
    includeDomains: z.array(z.string()).optional(),
    excludeDomains: z.array(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        maxResults: "max_results",
        searchPrompt: "search_prompt",
        includeDomains: "include_domains",
        excludeDomains: "exclude_domains",
    });
});
export function responsesRequestPluginWebToJSON(responsesRequestPluginWeb) {
    return JSON.stringify(ResponsesRequestPluginWeb$outboundSchema.parse(responsesRequestPluginWeb));
}
/** @internal */
export const ResponsesRequestPluginModeration$outboundSchema = z.object({
    id: z.literal("moderation"),
});
export function responsesRequestPluginModerationToJSON(responsesRequestPluginModeration) {
    return JSON.stringify(ResponsesRequestPluginModeration$outboundSchema.parse(responsesRequestPluginModeration));
}
/** @internal */
export const ResponsesRequestPluginAutoRouter$outboundSchema = z.object({
    id: z.literal("auto-router"),
    enabled: z.boolean().optional(),
    allowedModels: z.array(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        allowedModels: "allowed_models",
    });
});
export function responsesRequestPluginAutoRouterToJSON(responsesRequestPluginAutoRouter) {
    return JSON.stringify(ResponsesRequestPluginAutoRouter$outboundSchema.parse(responsesRequestPluginAutoRouter));
}
/** @internal */
export const ResponsesRequestPluginUnion$outboundSchema = z.union([
    z.lazy(() => ResponsesRequestPluginAutoRouter$outboundSchema),
    z.lazy(() => ResponsesRequestPluginModeration$outboundSchema),
    z.lazy(() => ResponsesRequestPluginWeb$outboundSchema),
    z.lazy(() => ResponsesRequestPluginFileParser$outboundSchema),
    z.lazy(() => ResponsesRequestPluginResponseHealing$outboundSchema),
    z.lazy(() => ResponsesRequestPluginContextCompression$outboundSchema),
]);
export function responsesRequestPluginUnionToJSON(responsesRequestPluginUnion) {
    return JSON.stringify(ResponsesRequestPluginUnion$outboundSchema.parse(responsesRequestPluginUnion));
}
/** @internal */
export const ResponsesRequestTrace$outboundSchema = z.object({
    traceId: z.string().optional(),
    traceName: z.string().optional(),
    spanName: z.string().optional(),
    generationName: z.string().optional(),
    parentSpanId: z.string().optional(),
    additionalProperties: z.record(z.string(), z.nullable(z.any())).optional(),
}).transform((v) => {
    return {
        ...v.additionalProperties,
        ...remap$(v, {
            traceId: "trace_id",
            traceName: "trace_name",
            spanName: "span_name",
            generationName: "generation_name",
            parentSpanId: "parent_span_id",
            additionalProperties: null,
        }),
    };
});
export function responsesRequestTraceToJSON(responsesRequestTrace) {
    return JSON.stringify(ResponsesRequestTrace$outboundSchema.parse(responsesRequestTrace));
}
/** @internal */
export const ResponsesRequest$outboundSchema = z.object({
    input: InputsUnion$outboundSchema.optional(),
    instructions: z.nullable(z.string()).optional(),
    metadata: z.nullable(z.record(z.string(), z.string())).optional(),
    tools: z.array(z.union([
        z.lazy(() => ResponsesRequestToolFunction$outboundSchema),
        PreviewWebSearchServerTool$outboundSchema,
        Preview20250311WebSearchServerTool$outboundSchema,
        LegacyWebSearchServerTool$outboundSchema,
        WebSearchServerTool$outboundSchema,
        FileSearchServerTool$outboundSchema,
        ComputerUseServerTool$outboundSchema,
        CodeInterpreterServerTool$outboundSchema,
        McpServerTool$outboundSchema,
        ImageGenerationServerTool$outboundSchema,
        CodexLocalShellTool$outboundSchema,
        ShellServerTool$outboundSchema,
        ApplyPatchServerTool$outboundSchema,
        CustomTool$outboundSchema,
        DatetimeServerTool$outboundSchema.and(z.object({ type: z.literal("openrouter:datetime") })),
        WebSearchServerToolOpenRouter$outboundSchema,
    ])).optional(),
    toolChoice: OpenAIResponsesToolChoiceUnion$outboundSchema.optional(),
    parallelToolCalls: z.nullable(z.boolean()).optional(),
    model: z.string().optional(),
    models: z.array(z.string()).optional(),
    text: TextExtendedConfig$outboundSchema.optional(),
    reasoning: z.nullable(ReasoningConfig$outboundSchema).optional(),
    maxOutputTokens: z.nullable(z.number()).optional(),
    temperature: z.nullable(z.number()).optional(),
    topP: z.nullable(z.number()).optional(),
    topLogprobs: z.nullable(z.int()).optional(),
    maxToolCalls: z.nullable(z.int()).optional(),
    presencePenalty: z.nullable(z.number()).optional(),
    frequencyPenalty: z.nullable(z.number()).optional(),
    topK: z.number().optional(),
    imageConfig: z.record(z.string(), z.union([z.string(), z.number()]))
        .optional(),
    modalities: z.array(OutputModalityEnum$outboundSchema).optional(),
    promptCacheKey: z.nullable(z.string()).optional(),
    previousResponseId: z.nullable(z.string()).optional(),
    prompt: z.nullable(StoredPromptTemplate$outboundSchema).optional(),
    include: z.nullable(z.array(ResponseIncludesEnum$outboundSchema)).optional(),
    background: z.nullable(z.boolean()).optional(),
    safetyIdentifier: z.nullable(z.string()).optional(),
    store: z.literal(false).default(false),
    serviceTier: z.nullable(ResponsesRequestServiceTier$outboundSchema.default("auto")),
    truncation: z.nullable(OpenAIResponsesTruncation$outboundSchema).optional(),
    stream: z.boolean().default(false),
    provider: z.nullable(z.lazy(() => ResponsesRequestProvider$outboundSchema))
        .optional(),
    plugins: z.array(z.union([
        z.lazy(() => ResponsesRequestPluginAutoRouter$outboundSchema),
        z.lazy(() => ResponsesRequestPluginModeration$outboundSchema),
        z.lazy(() => ResponsesRequestPluginWeb$outboundSchema),
        z.lazy(() => ResponsesRequestPluginFileParser$outboundSchema),
        z.lazy(() => ResponsesRequestPluginResponseHealing$outboundSchema),
        z.lazy(() => ResponsesRequestPluginContextCompression$outboundSchema),
    ])).optional(),
    user: z.string().optional(),
    sessionId: z.string().optional(),
    trace: z.lazy(() => ResponsesRequestTrace$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        toolChoice: "tool_choice",
        parallelToolCalls: "parallel_tool_calls",
        maxOutputTokens: "max_output_tokens",
        topP: "top_p",
        topLogprobs: "top_logprobs",
        maxToolCalls: "max_tool_calls",
        presencePenalty: "presence_penalty",
        frequencyPenalty: "frequency_penalty",
        topK: "top_k",
        imageConfig: "image_config",
        promptCacheKey: "prompt_cache_key",
        previousResponseId: "previous_response_id",
        safetyIdentifier: "safety_identifier",
        serviceTier: "service_tier",
        sessionId: "session_id",
    });
});
export function responsesRequestToJSON(responsesRequest) {
    return JSON.stringify(ResponsesRequest$outboundSchema.parse(responsesRequest));
}
