// @openrouter/sdk/esm/models/streamevents.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 4aa1487f961f
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { AnnotationAddedEvent$inboundSchema, } from "./annotationaddedevent.js";
import { ContentPartAddedEvent$inboundSchema, } from "./contentpartaddedevent.js";
import { ContentPartDoneEvent$inboundSchema, } from "./contentpartdoneevent.js";
import { ErrorEvent$inboundSchema } from "./errorevent.js";
import { FunctionCallArgsDeltaEvent$inboundSchema, } from "./functioncallargsdeltaevent.js";
import { FunctionCallArgsDoneEvent$inboundSchema, } from "./functioncallargsdoneevent.js";
import { ImageGenCallCompletedEvent$inboundSchema, } from "./imagegencallcompletedevent.js";
import { ImageGenCallGeneratingEvent$inboundSchema, } from "./imagegencallgeneratingevent.js";
import { ImageGenCallInProgressEvent$inboundSchema, } from "./imagegencallinprogressevent.js";
import { ImageGenCallPartialImageEvent$inboundSchema, } from "./imagegencallpartialimageevent.js";
import { OpenResponsesResult$inboundSchema, } from "./openresponsesresult.js";
import { OutputItems$inboundSchema } from "./outputitems.js";
import { ReasoningDeltaEvent$inboundSchema, } from "./reasoningdeltaevent.js";
import { ReasoningDoneEvent$inboundSchema, } from "./reasoningdoneevent.js";
import { ReasoningSummaryPartAddedEvent$inboundSchema, } from "./reasoningsummarypartaddedevent.js";
import { ReasoningSummaryPartDoneEvent$inboundSchema, } from "./reasoningsummarypartdoneevent.js";
import { ReasoningSummaryTextDeltaEvent$inboundSchema, } from "./reasoningsummarytextdeltaevent.js";
import { ReasoningSummaryTextDoneEvent$inboundSchema, } from "./reasoningsummarytextdoneevent.js";
import { RefusalDeltaEvent$inboundSchema, } from "./refusaldeltaevent.js";
import { RefusalDoneEvent$inboundSchema, } from "./refusaldoneevent.js";
import { TextDeltaEvent$inboundSchema, } from "./textdeltaevent.js";
import { TextDoneEvent$inboundSchema } from "./textdoneevent.js";
import { WebSearchCallCompletedEvent$inboundSchema, } from "./websearchcallcompletedevent.js";
import { WebSearchCallInProgressEvent$inboundSchema, } from "./websearchcallinprogressevent.js";
import { WebSearchCallSearchingEvent$inboundSchema, } from "./websearchcallsearchingevent.js";
/** @internal */
export const StreamEventsResponseOutputItemDone$inboundSchema = z.object({
    type: z.literal("response.output_item.done"),
    output_index: z.number(),
    item: OutputItems$inboundSchema,
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "output_index": "outputIndex",
        "sequence_number": "sequenceNumber",
    });
});
export function streamEventsResponseOutputItemDoneFromJSON(jsonString) {
    return safeParse(jsonString, (x) => StreamEventsResponseOutputItemDone$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'StreamEventsResponseOutputItemDone' from JSON`);
}
/** @internal */
export const StreamEventsResponseOutputItemAdded$inboundSchema = z.object({
    type: z.literal("response.output_item.added"),
    output_index: z.number(),
    item: OutputItems$inboundSchema,
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "output_index": "outputIndex",
        "sequence_number": "sequenceNumber",
    });
});
export function streamEventsResponseOutputItemAddedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => StreamEventsResponseOutputItemAdded$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'StreamEventsResponseOutputItemAdded' from JSON`);
}
/** @internal */
export const StreamEventsResponseFailed$inboundSchema = z.object({
    type: z.literal("response.failed"),
    response: OpenResponsesResult$inboundSchema,
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "sequence_number": "sequenceNumber",
    });
});
export function streamEventsResponseFailedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => StreamEventsResponseFailed$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'StreamEventsResponseFailed' from JSON`);
}
/** @internal */
export const StreamEventsResponseIncomplete$inboundSchema = z.object({
    type: z.literal("response.incomplete"),
    response: OpenResponsesResult$inboundSchema,
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "sequence_number": "sequenceNumber",
    });
});
export function streamEventsResponseIncompleteFromJSON(jsonString) {
    return safeParse(jsonString, (x) => StreamEventsResponseIncomplete$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'StreamEventsResponseIncomplete' from JSON`);
}
/** @internal */
export const StreamEventsResponseCompleted$inboundSchema = z.object({
    type: z.literal("response.completed"),
    response: OpenResponsesResult$inboundSchema,
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "sequence_number": "sequenceNumber",
    });
});
export function streamEventsResponseCompletedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => StreamEventsResponseCompleted$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'StreamEventsResponseCompleted' from JSON`);
}
/** @internal */
export const StreamEventsResponseInProgress$inboundSchema = z.object({
    type: z.literal("response.in_progress"),
    response: OpenResponsesResult$inboundSchema,
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "sequence_number": "sequenceNumber",
    });
});
export function streamEventsResponseInProgressFromJSON(jsonString) {
    return safeParse(jsonString, (x) => StreamEventsResponseInProgress$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'StreamEventsResponseInProgress' from JSON`);
}
/** @internal */
export const StreamEventsResponseCreated$inboundSchema = z.object({
    type: z.literal("response.created"),
    response: OpenResponsesResult$inboundSchema,
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "sequence_number": "sequenceNumber",
    });
});
export function streamEventsResponseCreatedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => StreamEventsResponseCreated$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'StreamEventsResponseCreated' from JSON`);
}
/** @internal */
export const StreamEvents$inboundSchema = z
    .union([
    z.lazy(() => StreamEventsResponseCreated$inboundSchema),
    z.lazy(() => StreamEventsResponseInProgress$inboundSchema),
    z.lazy(() => StreamEventsResponseCompleted$inboundSchema),
    z.lazy(() => StreamEventsResponseIncomplete$inboundSchema),
    z.lazy(() => StreamEventsResponseFailed$inboundSchema),
    ErrorEvent$inboundSchema,
    z.lazy(() => StreamEventsResponseOutputItemAdded$inboundSchema),
    z.lazy(() => StreamEventsResponseOutputItemDone$inboundSchema),
    ContentPartAddedEvent$inboundSchema,
    ContentPartDoneEvent$inboundSchema,
    TextDeltaEvent$inboundSchema,
    TextDoneEvent$inboundSchema,
    RefusalDeltaEvent$inboundSchema,
    RefusalDoneEvent$inboundSchema,
    AnnotationAddedEvent$inboundSchema,
    FunctionCallArgsDeltaEvent$inboundSchema,
    FunctionCallArgsDoneEvent$inboundSchema,
    ReasoningDeltaEvent$inboundSchema,
    ReasoningDoneEvent$inboundSchema,
    ReasoningSummaryPartAddedEvent$inboundSchema,
    ReasoningSummaryPartDoneEvent$inboundSchema,
    ReasoningSummaryTextDeltaEvent$inboundSchema,
    ReasoningSummaryTextDoneEvent$inboundSchema,
    ImageGenCallInProgressEvent$inboundSchema,
    ImageGenCallGeneratingEvent$inboundSchema,
    ImageGenCallPartialImageEvent$inboundSchema,
    ImageGenCallCompletedEvent$inboundSchema,
    WebSearchCallInProgressEvent$inboundSchema,
    WebSearchCallSearchingEvent$inboundSchema,
    WebSearchCallCompletedEvent$inboundSchema,
]);
export function streamEventsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => StreamEvents$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'StreamEvents' from JSON`);
}
