// @openrouter/sdk/esm/models/textdeltaevent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 920ea48e0f90
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const TextDeltaEventTopLogprob$inboundSchema = z.object({
    token: z.string().optional(),
    logprob: z.number().optional(),
    bytes: z.array(z.number()).optional(),
});
export function textDeltaEventTopLogprobFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TextDeltaEventTopLogprob$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TextDeltaEventTopLogprob' from JSON`);
}
/** @internal */
export const TextDeltaEventLogprob$inboundSchema = z.object({
    logprob: z.number(),
    token: z.string(),
    top_logprobs: z.array(z.lazy(() => TextDeltaEventTopLogprob$inboundSchema))
        .optional(),
    bytes: z.array(z.number()).optional(),
}).transform((v) => {
    return remap$(v, {
        "top_logprobs": "topLogprobs",
    });
});
export function textDeltaEventLogprobFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TextDeltaEventLogprob$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TextDeltaEventLogprob' from JSON`);
}
/** @internal */
export const TextDeltaEvent$inboundSchema = z.object({
    type: z.literal("response.output_text.delta"),
    logprobs: z.array(z.lazy(() => TextDeltaEventLogprob$inboundSchema)),
    output_index: z.number(),
    item_id: z.string(),
    content_index: z.number(),
    delta: z.string(),
    sequence_number: z.number(),
}).transform((v) => {
    return remap$(v, {
        "output_index": "outputIndex",
        "item_id": "itemId",
        "content_index": "contentIndex",
        "sequence_number": "sequenceNumber",
    });
});
export function textDeltaEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TextDeltaEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TextDeltaEvent' from JSON`);
}
