// @openrouter/sdk/esm/models/textdoneevent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 006e3645c287
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const TextDoneEventTopLogprob$inboundSchema = z.object({
    token: z.string().optional(),
    logprob: z.number().optional(),
    bytes: z.array(z.number()).optional(),
});
export function textDoneEventTopLogprobFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TextDoneEventTopLogprob$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TextDoneEventTopLogprob' from JSON`);
}
/** @internal */
export const TextDoneEventLogprob$inboundSchema = z.object({
    logprob: z.number(),
    token: z.string(),
    top_logprobs: z.array(z.lazy(() => TextDoneEventTopLogprob$inboundSchema))
        .optional(),
    bytes: z.array(z.number()).optional(),
}).transform((v) => {
    return remap$(v, {
        "top_logprobs": "topLogprobs",
    });
});
export function textDoneEventLogprobFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TextDoneEventLogprob$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TextDoneEventLogprob' from JSON`);
}
/** @internal */
export const TextDoneEvent$inboundSchema = z
    .object({
    type: z.literal("response.output_text.done"),
    output_index: z.number(),
    item_id: z.string(),
    content_index: z.number(),
    text: z.string(),
    sequence_number: z.number(),
    logprobs: z.array(z.lazy(() => TextDoneEventLogprob$inboundSchema)),
}).transform((v) => {
    return remap$(v, {
        "output_index": "outputIndex",
        "item_id": "itemId",
        "content_index": "contentIndex",
        "sequence_number": "sequenceNumber",
    });
});
export function textDoneEventFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TextDoneEvent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TextDoneEvent' from JSON`);
}
