// @openrouter/sdk/esm/models/websearchservertoolopenrouter.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 99dffefc1c11
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
/** @internal */
export const WebSearchServerToolOpenRouterParameters$outboundSchema = z.object({
    maxResults: z.number().optional(),
    maxTotalResults: z.number().optional(),
}).transform((v) => {
    return remap$(v, {
        maxResults: "max_results",
        maxTotalResults: "max_total_results",
    });
});
export function webSearchServerToolOpenRouterParametersToJSON(webSearchServerToolOpenRouterParameters) {
    return JSON.stringify(WebSearchServerToolOpenRouterParameters$outboundSchema.parse(webSearchServerToolOpenRouterParameters));
}
/** @internal */
export const WebSearchServerToolOpenRouter$outboundSchema = z.object({
    type: z.literal("openrouter:web_search"),
    parameters: z.lazy(() => WebSearchServerToolOpenRouterParameters$outboundSchema).optional(),
});
export function webSearchServerToolOpenRouterToJSON(webSearchServerToolOpenRouter) {
    return JSON.stringify(WebSearchServerToolOpenRouter$outboundSchema.parse(webSearchServerToolOpenRouter));
}
