// @openrouter/sdk/esm/models/websearchservertool.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 478da2f8e455
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as openEnums from "../types/enums.js";
import { SearchContextSizeEnum$inboundSchema, SearchContextSizeEnum$outboundSchema, } from "./searchcontextsizeenum.js";
import { WebSearchUserLocation$inboundSchema, WebSearchUserLocation$outboundSchema, } from "./websearchuserlocation.js";
/**
 * Which search engine to use. "auto" (default) uses native if the provider supports it, otherwise Exa. "native" forces the provider's built-in search. "exa" forces the Exa search API. "firecrawl" uses Firecrawl (requires BYOK). "parallel" uses the Parallel search API.
 */
export const WebSearchServerToolEngine = {
    Auto: "auto",
    Native: "native",
    Exa: "exa",
    Firecrawl: "firecrawl",
    Parallel: "parallel",
};
/** @internal */
export const WebSearchServerToolFilters$inboundSchema = z.object({
    allowed_domains: z.nullable(z.array(z.string())).optional(),
    excluded_domains: z.nullable(z.array(z.string())).optional(),
}).transform((v) => {
    return remap$(v, {
        "allowed_domains": "allowedDomains",
        "excluded_domains": "excludedDomains",
    });
});
/** @internal */
export const WebSearchServerToolFilters$outboundSchema = z.object({
    allowedDomains: z.nullable(z.array(z.string())).optional(),
    excludedDomains: z.nullable(z.array(z.string())).optional(),
}).transform((v) => {
    return remap$(v, {
        allowedDomains: "allowed_domains",
        excludedDomains: "excluded_domains",
    });
});
export function webSearchServerToolFiltersToJSON(webSearchServerToolFilters) {
    return JSON.stringify(WebSearchServerToolFilters$outboundSchema.parse(webSearchServerToolFilters));
}
export function webSearchServerToolFiltersFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebSearchServerToolFilters$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebSearchServerToolFilters' from JSON`);
}
/** @internal */
export const WebSearchServerToolEngine$inboundSchema = openEnums.inboundSchema(WebSearchServerToolEngine);
/** @internal */
export const WebSearchServerToolEngine$outboundSchema = openEnums.outboundSchema(WebSearchServerToolEngine);
/** @internal */
export const WebSearchServerTool$inboundSchema = z.object({
    type: z.literal("web_search_2025_08_26"),
    filters: z.nullable(z.lazy(() => WebSearchServerToolFilters$inboundSchema))
        .optional(),
    search_context_size: SearchContextSizeEnum$inboundSchema.optional(),
    user_location: z.nullable(WebSearchUserLocation$inboundSchema).optional(),
    engine: WebSearchServerToolEngine$inboundSchema.optional(),
    max_results: z.number().optional(),
}).transform((v) => {
    return remap$(v, {
        "search_context_size": "searchContextSize",
        "user_location": "userLocation",
        "max_results": "maxResults",
    });
});
/** @internal */
export const WebSearchServerTool$outboundSchema = z.object({
    type: z.literal("web_search_2025_08_26"),
    filters: z.nullable(z.lazy(() => WebSearchServerToolFilters$outboundSchema))
        .optional(),
    searchContextSize: SearchContextSizeEnum$outboundSchema.optional(),
    userLocation: z.nullable(WebSearchUserLocation$outboundSchema).optional(),
    engine: WebSearchServerToolEngine$outboundSchema.optional(),
    maxResults: z.number().optional(),
}).transform((v) => {
    return remap$(v, {
        searchContextSize: "search_context_size",
        userLocation: "user_location",
        maxResults: "max_results",
    });
});
export function webSearchServerToolToJSON(webSearchServerTool) {
    return JSON.stringify(WebSearchServerTool$outboundSchema.parse(webSearchServerTool));
}
export function webSearchServerToolFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebSearchServerTool$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebSearchServerTool' from JSON`);
}
