// @openrouter/sdk/esm/models/websearchuserlocation.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: cc946a3064e3
 */
import * as z from "zod/v4";
import { safeParse } from "../lib/schemas.js";
export const WebSearchUserLocationType = {
    Approximate: "approximate",
};
/** @internal */
export const WebSearchUserLocationType$inboundSchema = z.enum(WebSearchUserLocationType);
/** @internal */
export const WebSearchUserLocationType$outboundSchema = WebSearchUserLocationType$inboundSchema;
/** @internal */
export const WebSearchUserLocation$inboundSchema = z.object({
    type: WebSearchUserLocationType$inboundSchema.optional(),
    city: z.nullable(z.string()).optional(),
    country: z.nullable(z.string()).optional(),
    region: z.nullable(z.string()).optional(),
    timezone: z.nullable(z.string()).optional(),
});
/** @internal */
export const WebSearchUserLocation$outboundSchema = z.object({
    type: WebSearchUserLocationType$outboundSchema.optional(),
    city: z.nullable(z.string()).optional(),
    country: z.nullable(z.string()).optional(),
    region: z.nullable(z.string()).optional(),
    timezone: z.nullable(z.string()).optional(),
});
export function webSearchUserLocationToJSON(webSearchUserLocation) {
    return JSON.stringify(WebSearchUserLocation$outboundSchema.parse(webSearchUserLocation));
}
export function webSearchUserLocationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => WebSearchUserLocation$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebSearchUserLocation' from JSON`);
}
